*** original_scripts/mortol/gml_Object_o29_Player_Step_2.gml	Sun May 11 20:25:31 2025
--- ufo50_modded_scripts/mortol/gml_Object_o29_Player_Step_2.gml	Sun May 11 20:14:00 2025
***************
*** 56 ****
--- 56,86 ----
  }
+ 
+ if (controller == 3)
+ {
+     if (sprite_index == s29_Player)
+         sprite_index = s29_Player3;
+ 
+     if (sprite_index == s29_PlayerJumping)
+         sprite_index = s29_PlayerJumping3;
+     
+     if (sprite_index == s29_PlayerSwimming)
+         sprite_index = s29_PlayerSwimming3;
+ 
+     if (sprite_index == s29_PlayerClimbing)
+         sprite_index = s29_PlayerClimbing3;
+ } 
+ 
+ if (controller == 4)
+ {
+     if (sprite_index == s29_Player)
+         sprite_index = s29_Player4;
+ 
+     if (sprite_index == s29_PlayerJumping)
+         sprite_index = s29_PlayerJumping4;
+     
+     if (sprite_index == s29_PlayerSwimming)
+         sprite_index = s29_PlayerSwimming4;
+ 
+     if (sprite_index == s29_PlayerClimbing)
+         sprite_index = s29_PlayerClimbing4;
+ }
