*** original_scripts/mortol/gml_Object_o29_Torpedo_Step_2.gml	Fri May  9 15:27:25 2025
--- ufo50_modded_scripts/mortol/gml_Object_o29_Torpedo_Step_2.gml	Sun Apr 27 20:58:36 2025
***************
*** 6,7 ****
--- 6,11 ----
          sprite_index = s29_Torpedo2;
+     if (o29_Game.currPlayer == 2)
+         sprite_index = s29_Torpedo3;
+     if (o29_Game.currPlayer == 3)
+         sprite_index = s29_Torpedo4;
  }
