*** ufo50_original_scripts/mortol/gml_Object_o29_Player_Step_0.gml	Fri May  9 18:27:35 2025
--- ufo50_modded_scripts/mortol/gml_Object_o29_Player_Step_0.gml	Sun Apr 27 20:58:36 2025
***************
*** 315,316 ****
--- 315,320 ----
              dyingPlayer.sprite_index = s29_PlayerDie2;
+         if (o29_Game.currPlayer == 2)
+             dyingPlayer.sprite_index = s29_PlayerDie3;
+         if (o29_Game.currPlayer == 3)
+             dyingPlayer.sprite_index = s29_PlayerDie4;
          
