*** original_scripts/mortol/gml_Object_o29_Game_Other_21.gml	Fri May  9 19:27:30 2025
--- ufo50_modded_scripts/mortol/gml_Object_o29_Game_Other_21.gml	Fri May  9 19:57:07 2025
***************
*** 53,58 ****
          
!         if (numPlayers == 1)
!             currPlayer = 0;
!         else
!             currPlayer = 1;
          
--- 53,55 ----
          
!         currPlayer = numPlayers - 1;
          
