*** original_scripts/mortol/gml_Object_o29_Torpedo_Step_0.gml	Mon Sep  8 19:19:28 2025
--- ufo50_modded_scripts/mortol/gml_Object_o29_Torpedo_Step_0.gml	Tue Sep  9 21:38:46 2025
***************
*** 205,206 ****
--- 205,210 ----
              tc.sprite_index = s29_TorpedoCrashed2;
+         if (o29_Game.currPlayer == 2)
+             tc.sprite_index = s29_TorpedoCrashed3;
+         if (o29_Game.currPlayer == 3)
+             tc.sprite_index = s29_TorpedoCrashed4;
      }
