*** original_scripts/mortol/gml_Object_o29_Player_Step_0.gml	Mon Sep  8 19:19:28 2025
--- ufo50_modded_scripts/mortol/gml_Object_o29_Player_Step_0.gml	Tue Sep  9 21:37:27 2025
***************
*** 320,321 ****
--- 320,325 ----
              dyingPlayer.sprite_index = s29_PlayerDie2;
+         if (o29_Game.currPlayer == 2)
+             dyingPlayer.sprite_index = s29_PlayerDie3;
+         if (o29_Game.currPlayer == 3)
+             dyingPlayer.sprite_index = s29_PlayerDie4;
          
