*** original_scripts/mortol/gml_Object_o29_Game_Other_13.gml	Sun Apr 27 20:54:41 2025
--- ufo50_modded_scripts/mortol/gml_Object_o29_Game_Other_13.gml	Sun Apr 27 20:58:36 2025
***************
*** 65,68 ****
              
!             if (numPlayers == 2)
!                 currPlayer = !currPlayer;
              
--- 65,71 ----
              
!             if (currPlayer == numPlayers - 1) {
!                 currPlayer = 0
!             } else {
!                 currPlayer += 1;
!             }
              
***************
*** 149,152 ****
                  
!                 if (numPlayers == 2)
!                     currPlayer = !currPlayer;
                  
--- 152,158 ----
                  
!                 if (currPlayer == numPlayers - 1) {
!                     currPlayer = 0
!                 } else {
!                     currPlayer += 1;
!                 }
                  
