*** original_scripts/mortol/gml_Object_o29_PlayerParachute_Step_0.gml	Thu May 15 21:38:36 2025
--- ufo50_modded_scripts/mortol/gml_Object_o29_PlayerParachute_Step_0.gml	Sun May 11 20:14:00 2025
***************
*** 53,54 ****
--- 53,58 ----
          para.sprite_index = s29_Parachute2;
+     if (controller == 3)
+         para.sprite_index = s29_Parachute3;
+     if (controller == 4)
+         para.sprite_index = s29_Parachute4;
      
