*** original_scripts/mortol/gml_GlobalScript_scr29_SaveGame.gml	Sun May 11 20:13:18 2025
--- ufo50_modded_scripts/mortol/gml_GlobalScript_scr29_SaveGame.gml	Fri May  9 20:16:24 2025
***************
*** 18,20 ****
              
!             for (var p = 1; p <= 2; p++)
              {
--- 18,20 ----
              
!             for (var p = 1; p <= 4; p++)
              {
***************
*** 111,113 ****
          {
!             for (var p = 1; p <= 2; p++)
              {
--- 111,113 ----
          {
!             for (var p = 1; p <= 4; p++)
              {
