*** original_scripts/mortol/gml_Object_o29_Game_Other_16.gml	Sun May 11 20:25:31 2025
--- ufo50_modded_scripts/mortol/gml_Object_o29_Game_Other_16.gml	Sun May 11 20:14:00 2025
***************
*** 310,315 ****
      {
!         if (numPlayers == 1)
!             currPlayer = 0;
!         else
!             currPlayer = 1;
          
--- 310,312 ----
      {
!         currPlayer = numPlayers - 1;
          
