*** original_scripts/mortol/gml_Object_o29_PlayerParachute_Step_0.gml	Mon Sep  8 19:19:28 2025
--- ufo50_modded_scripts/mortol/gml_Object_o29_PlayerParachute_Step_0.gml	Tue Sep  9 21:38:09 2025
***************
*** 55,56 ****
--- 55,60 ----
          para.sprite_index = s29_Parachute2;
+     if (controller == 3)
+         para.sprite_index = s29_Parachute3;
+     if (controller == 4)
+         para.sprite_index = s29_Parachute4;
      
