*** original_scripts/party_house/gml_Object_o36_Game_Other_15.gml	Fri Sep 12 15:29:48 2025
--- ufo50_modded_scripts/party_house/gml_Object_o36_Game_Other_15.gml	Fri Sep 12 15:31:09 2025
***************
*** 51,63 ****
      
!     if ((wonGame[0] || wonGame[1]) && phase == PHASE_SCORING)
      {
!         scr36_ChangeState(STATE_WIN);
!         exit;
!     }
!     else
!     {
!         substate = 2;
!         listPos = scoreStart;
!         stateTimer = 0;
      }
  }
--- 51,63 ----
      
!     for (var op = 0; op < numPlayers; op++)
      {
!         if (wonGame[op] && phase == PHASE_SCORING)
!         {
!             scr36_ChangeState(STATE_WIN);
!             exit;
!         }
      }
+     substate = 2;
+     listPos = scoreStart;
+     stateTimer = 0;
  }
