*** original_scripts/party_house/gml_Object_o36_Game_Other_16.gml	Thu Aug 28 19:59:01 2025
--- ufo50_modded_scripts/party_house/gml_Object_o36_Game_Other_16.gml	Sat Aug 23 18:45:14 2025
***************
*** 17,19 ****
          }
!         else if (numPlayers == 2 && wonGame[0])
          {
--- 17,19 ----
          }
!         else if (numPlayers >= 2 && array_any(wonGame, function(_val, ind){ return _val; }))
          {
