*** original_scripts/party_house/gml_GlobalScript_scr36_SaveGame.gml	Thu Aug 28 19:59:01 2025
--- ufo50_modded_scripts/party_house/gml_GlobalScript_scr36_SaveGame.gml	Thu Aug 28 19:20:50 2025
***************
*** 29,32 ****
                  scrWriteReal("scenarioWins" + string(i + 1), scenarioWins[i]);
!                 scrWriteReal("scenarioWinsP1" + string(i + 1), scenarioWinsP1[i]);
!                 scrWriteReal("scenarioWinsP2" + string(i + 1), scenarioWinsP2[i]);
              }
--- 29,32 ----
                  scrWriteReal("scenarioWins" + string(i + 1), scenarioWins[i]);
!                 for (var player = 0; player < 4; player++)
!                     scrWriteReal("scenarioWinsMulti" + string(player + 1) + string(i + 1), scenarioWinsMulti[player][i]);
              }
***************
*** 82,85 ****
                  scenarioWins[i] = scrReadReal("scenarioWins" + string(i + 1), 0);
!                 scenarioWinsP1[i] = scrReadReal("scenarioWinsP1" + string(i + 1), 0);
!                 scenarioWinsP2[i] = scrReadReal("scenarioWinsP2" + string(i + 1), 0);
              }
--- 82,86 ----
                  scenarioWins[i] = scrReadReal("scenarioWins" + string(i + 1), 0);
! 
!                 for (var player = 0; player < 4; player++)
!                     scenarioWinsMulti[player][i] = scrReadReal("scenarioWinsMulti" + string(player + 1) + string (i + 1), 0);
              }
