*** original_scripts/waldorf/gml_Object_o21_Game_Create_0.gml	Fri May 16 21:04:41 2025
--- ufo50_modded_scripts/waldorf/gml_Object_o21_Game_Create_0.gml	Fri May 16 20:06:42 2025
***************
*** 19,20 ****
--- 19,22 ----
  P2_WIN = 2;
+ P3_WIN = 3;
+ P4_WIN = 4;
  CAMERA_RADIUS = 90;
***************
*** 46,47 ****
--- 48,58 ----
      state = STATE_WAR_SETUP;
+     livePlayers = [1, 2];
+ 
+ if (numPlayers == 3)
+     state = STATE_WAR_SETUP;
+     livePlayers = [1, 2, 3];
+ 
+ if (numPlayers == 4)
+     state = STATE_WAR_SETUP;
+     livePlayers = [1, 2, 3, 4];
  
