*** original_scripts/bushido_ball/gml_Object_o22_Game_Other_16.gml	Tue Oct 14 20:52:30 2025
--- ufo50_modded_scripts/bushido_ball/gml_Object_o22_Game_Other_16.gml	Thu Oct 16 08:13:38 2025
***************
*** 48,50 ****
          }
!         else if (numPlayers == 4)
          {
--- 48,50 ----
          }
!         else if (numPlayers == 4 || numPlayers == 5)
          {
***************
*** 62,63 ****
      }
! }
--- 62,63 ----
      }
! }
\ No newline at end of file
