*** original_scripts/bushido_ball/gml_Object_o22_HandRight_Step_0.gml	Fri Nov  7 20:49:38 2025
--- ufo50_modded_scripts/bushido_ball/gml_Object_o22_HandRight_Step_0.gml	Thu Oct 16 08:16:24 2025
***************
*** 5,7 ****
  {
!     scrGetInput(1);
      
--- 5,7 ----
  {
!     scrGetInput(o22_Game.controllerMap[1]);
      
***************
*** 112,114 ****
--- 112,119 ----
      o22_CharselRight.val = sel;
+ 
+     if (o22_Game.numPlayers == 5)
+         o22_CharselRight.y = 300;
+ 
      o22_PortraitRight.val = sel;
+     
      var _xv = camera_get_view_x(view_get_camera(0));
***************
*** 167,175 ****
  {
!     var _canBackOut = true;
      
!     with (o22_HandLeft)
!     {
!         if (ready)
!             _canBackOut = false;
!     }
      
--- 172,177 ----
  {
!     var _canBackOut = !o22_HandLeft.ready;
      
!     if (o22_Game.numPlayers == 5)
!         _canBackOut = _canBackOut || !o22_HandThree.ready || !o22_HandFour.ready;
      
