*** original_scripts/bushido_ball/gml_Object_o22_Game_Create_0.gml	Fri Jun  6 20:57:21 2025
--- ufo50_modded_scripts/bushido_ball/gml_Object_o22_Game_Create_0.gml	Fri Jun  6 20:33:21 2025
***************
*** 20,21 ****
--- 20,26 ----
  numPlayers = global.numPlayers;
+ var playerList = [0, 1, 2, 3];
+ if (global.g22_randomiseControllers && global.numPlayers >= 4)
+     controllerMap = rng_random_array_shuffle(playerList);
+ else
+     controllerMap = playerList;
  
***************
*** 31,32 ****
--- 36,39 ----
  p2_sel = -1;
+ p3_sel = -1;
+ p4_sel = -1;
  c1_sel = -1;
