*** original_scripts/bushido_ball/gml_Object_o22_Game_Other_13.gml	Tue Oct 14 20:52:30 2025
--- ufo50_modded_scripts/bushido_ball/gml_Object_o22_Game_Other_13.gml	Thu Oct 16 08:12:27 2025
***************
*** 136,138 ****
          }
!         else if (o22_Ball.x > (camera_get_view_x(view_get_camera(0)) + 384 + 16))
          {
--- 136,138 ----
          }
!         else if (o22_Ball.x > (camera_get_view_x(view_get_camera(0)) + __view_get(e__VW.WView, 0) + 16))
          {
***************
*** 141,143 ****
          }
!         else if (o22_Ball.y < camera_get_view_y(view_get_camera(0)) || o22_Ball.y > (camera_get_view_y(view_get_camera(0)) + 216 + 16))
          {
--- 141,143 ----
          }
!         else if (o22_Ball.y < camera_get_view_y(view_get_camera(0)) || o22_Ball.y > (camera_get_view_y(view_get_camera(0)) + __view_get(e__VW.HView, 0) + 16))
          {
***************
*** 185,187 ****
              {
!                 if (numPlayers != 4)
                  {
--- 185,187 ----
              {
!                 if (numPlayers != 4 && numPlayers != 5)
                  {
