*** original_scripts/bushido_ball/gml_Object_o22_Game_Draw_0.gml	Mon May 26 11:39:34 2025
--- ufo50_modded_scripts/bushido_ball/gml_Object_o22_Game_Draw_0.gml	Mon May 26 11:19:26 2025
***************
*** 189,191 ****
      
!     if (numPlayers == 2)
      {
--- 189,191 ----
      
!     if (numPlayers == 2 || numPlayers == 5)
      {
***************
*** 203,205 ****
                  draw_sprite(s22_CrossOut, 0, 128 + (48 * i), 304 + (48 * j));
!             else if (p1_sel == ii || p2_sel == ii)
                  draw_sprite(s22_CircleOut, 0, 128 + (48 * i), 304 + (48 * j));
--- 203,205 ----
                  draw_sprite(s22_CrossOut, 0, 128 + (48 * i), 304 + (48 * j));
!             else if (p1_sel == ii || p2_sel == ii || p3_sel == ii || p4_sel == ii)
                  draw_sprite(s22_CircleOut, 0, 128 + (48 * i), 304 + (48 * j));
***************
*** 335,337 ****
                  
!                 if (numPlayers == 4)
                      str = scrString("team") + " " + str;
--- 335,337 ----
                  
!                 if (numPlayers == 4 || numPlayers == 5)
                      str = scrString("team") + " " + str;
***************
*** 344,346 ****
                  
!                 if (numPlayers == 4)
                      str = scrString("team") + " " + str;
--- 344,346 ----
                  
!                 if (numPlayers == 4 || numPlayers == 5)
                      str = scrString("team") + " " + str;
***************
*** 389,391 ****
              
!             if (numPlayers == 4)
              {
--- 389,391 ----
              
!             if (numPlayers == 4 || numPlayers == 5)
              {
***************
*** 420,422 ****
              
!             if (numPlayers == 4)
              {
--- 420,422 ----
              
!             if (numPlayers == 4 || numPlayers == 5)
              {
***************
*** 476,478 ****
          
!         if (numPlayers == 4)
          {
--- 476,478 ----
          
!         if (numPlayers == 4 || numPlayers == 5)
          {
