*** original_scripts/bushido_ball/gml_Object_o22_HandRight_Step_0.gml	Fri Jun  6 20:57:22 2025
--- ufo50_modded_scripts/bushido_ball/gml_Object_o22_HandRight_Step_0.gml	Fri Jun  6 20:40:10 2025
***************
*** 5,7 ****
  {
!     scrGetInput(1);
      
--- 5,7 ----
  {
!     scrGetInput(o22_Game.controllerMap[1]);
      
***************
*** 165,173 ****
  {
!     var _canBackOut = true;
      
!     with (o22_HandLeft)
!     {
!         if (ready)
!             _canBackOut = false;
!     }
      
--- 165,170 ----
  {
!     var _canBackOut = !o22_HandLeft.ready;
      
!     if (o22_Game.numPlayers == 5)
!         _canBackOut = _canBackOut || !o22_HandThree.ready || !o22_HandFour.ready;
      
