*** original_scripts/bushido_ball/gml_Object_o22_Game_Draw_0.gml	Sat Jun  7 21:06:07 2025
--- ufo50_modded_scripts/bushido_ball/gml_Object_o22_Game_Draw_0.gml	Thu Jun 19 21:53:54 2025
***************
*** 5,7 ****
      draw_set_color(global.palette[0]);
!     draw_rectangle(scrViewX() + 118, scrViewY() + 32, scrViewX() + 264, scrViewY() + 160, 0);
      var _x = scrViewX() + 128;
--- 5,7 ----
      draw_set_color(global.palette[0]);
!     draw_rectangle(scrViewX() + 118, scrViewY() + 32, scrViewX() + 300, scrViewY() + 200, 0);
      var _x = scrViewX() + 128;
***************
*** 35,37 ****
      
!     _optionname[4] = scrString("option_back");
      
--- 35,42 ----
      
!     if (global.g22_randomiseControllers)
!         _optionname[4] = "CONTROLLERS: RANDOM";
!     else
!         _optionname[4] = "CONTROLLERS: FIXED";
! 
!     _optionname[5] = scrString("option_back");
      
***************
*** 45,47 ****
  }
! else if (state == STATE_CHAR_SELECT)
  {
--- 50,52 ----
  }
! else if (state == STATE_CHAR_SELECT && numPlayers <= 4)
  {
***************
*** 189,191 ****
      
!     if (numPlayers == 2)
      {
--- 194,196 ----
      
!     if (numPlayers == 2 || numPlayers == 5)
      {
***************
*** 203,205 ****
                  draw_sprite(s22_CrossOut, 0, 128 + (48 * i), 304 + (48 * j));
!             else if (p1_sel == ii || p2_sel == ii)
                  draw_sprite(s22_CircleOut, 0, 128 + (48 * i), 304 + (48 * j));
--- 208,210 ----
                  draw_sprite(s22_CrossOut, 0, 128 + (48 * i), 304 + (48 * j));
!             else if (p1_sel == ii || p2_sel == ii || p3_sel == ii || p4_sel == ii)
                  draw_sprite(s22_CircleOut, 0, 128 + (48 * i), 304 + (48 * j));
***************
*** 208,209 ****
--- 213,500 ----
  }
+ else if (state == STATE_CHAR_SELECT && numPlayers == 5)
+ {
+     var _stats;
+     var UPPER_OFFSET = 250;
+     var LOWER_OFFSET = 400;
+     _stats[0] = 2;
+     _stats[1] = 2;
+     _stats[2] = 2;
+     
+     switch (o22_HandLeft.sel)
+     {
+         case 0:
+             str = scrString("name_kotaro");
+             _stats[2] = 3;
+             break;
+         
+         case 1:
+             str = scrString("name_ayumi");
+             _stats[0] = 3;
+             break;
+         
+         case 2:
+             str = scrString("name_tomoe");
+             _stats[1] = 3;
+             break;
+         
+         case 3:
+             str = scrString("name_raizo");
+             _stats[1] = 1;
+             _stats[2] = 3;
+             break;
+         
+         case 4:
+             str = scrString("name_chiyome");
+             _stats[0] = 3;
+             _stats[2] = 1;
+             break;
+         
+         case 5:
+             str = scrString("name_yamada");
+             _stats[0] = 1;
+             _stats[1] = 3;
+             break;
+     }
+     
+     scrSetFont(global.fontTall);
+     var str = string_even(str, 3);
+     scrDrawTextCenteredPoint(str, 56, UPPER_OFFSET, 8);
+     scrSetFont(global.fontDefault);
+     
+     for (var _a = 0; _a < _stats[0]; _a++)
+         draw_sprite_ext(sFX_StarRotate, 0, ((scrViewX() + (__view_get(e__VW.WView, 0) * 0.5)) - 64) + (_a * 8), UPPER_OFFSET + 4, 1.2, 1, 0, c_black, 1);
+     
+     for (var _a = 0; _a < _stats[1]; _a++)
+         draw_sprite_ext(sFX_StarRotate, 0, ((scrViewX() + (__view_get(e__VW.WView, 0) * 0.5)) - 64) + (_a * 8), UPPER_OFFSET + 12, 1.2, 1, 0, c_black, 1);
+     
+     for (var _a = 0; _a < _stats[2]; _a++)
+         draw_sprite_ext(sFX_StarRotate, 0, ((scrViewX() + (__view_get(e__VW.WView, 0) * 0.5)) - 64) + (_a * 8), UPPER_OFFSET + 20, 1.2, 1, 0, c_black, 1);
+     
+     for (var _a = 0; _a < _stats[0]; _a++)
+         draw_sprite(sFX_StarRotate, 0, ((scrViewX() + (__view_get(e__VW.WView, 0) * 0.5)) - 64) + (_a * 8), UPPER_OFFSET + 3);
+     
+     for (var _a = 0; _a < _stats[1]; _a++)
+         draw_sprite(sFX_StarRotate, 0, ((scrViewX() + (__view_get(e__VW.WView, 0) * 0.5)) - 64) + (_a * 8), UPPER_OFFSET + 11);
+     
+     for (var _a = 0; _a < _stats[2]; _a++)
+         draw_sprite(sFX_StarRotate, 0, ((scrViewX() + (__view_get(e__VW.WView, 0) * 0.5)) - 64) + (_a * 8), UPPER_OFFSET + 19);
+     
+     if (instance_exists(o22_HandRight))
+     {
+         if (o22_HandRight.active)
+         {
+             _stats[0] = 2;
+             _stats[1] = 2;
+             _stats[2] = 2;
+             
+             switch (o22_HandRight.sel)
+             {
+                 case 0:
+                     str = scrString("name_kotaro");
+                     _stats[2] = 3;
+                     break;
+                 
+                 case 1:
+                     str = scrString("name_ayumi");
+                     _stats[0] = 3;
+                     break;
+                 
+                 case 2:
+                     str = scrString("name_tomoe");
+                     _stats[1] = 3;
+                     break;
+                 
+                 case 3:
+                     str = scrString("name_raizo");
+                     _stats[1] = 1;
+                     _stats[2] = 3;
+                     break;
+                 
+                 case 4:
+                     str = scrString("name_chiyome");
+                     _stats[0] = 3;
+                     _stats[2] = 1;
+                     break;
+                 
+                 case 5:
+                     str = scrString("name_yamada");
+                     _stats[0] = 1;
+                     _stats[1] = 3;
+                     break;
+             }
+             
+             scrSetFont(global.fontTall);
+             str = string_even(str, 3);
+             scrDrawTextCenteredPoint(str, 328, UPPER_OFFSET, 8);
+             scrSetFont(global.fontDefault);
+             
+             for (var _a = 0; _a < _stats[0]; _a++)
+                 draw_sprite_ext(sFX_StarRotate, 0, (scrViewX() + (__view_get(e__VW.WView, 0) * 0.5) + 64) - (_a * 8), UPPER_OFFSET + 4, 1.2, 1, 0, c_black, 1);
+             
+             for (var _a = 0; _a < _stats[1]; _a++)
+                 draw_sprite_ext(sFX_StarRotate, 0, (scrViewX() + (__view_get(e__VW.WView, 0) * 0.5) + 64) - (_a * 8), UPPER_OFFSET + 12, 1.2, 1, 0, c_black, 1);
+             
+             for (var _a = 0; _a < _stats[2]; _a++)
+                 draw_sprite_ext(sFX_StarRotate, 0, (scrViewX() + (__view_get(e__VW.WView, 0) * 0.5) + 64) - (_a * 8), UPPER_OFFSET + 20, 1.2, 1, 0, c_black, 1);
+             
+             for (var _a = 0; _a < _stats[0]; _a++)
+                 draw_sprite(sFX_StarRotate, 0, (scrViewX() + (__view_get(e__VW.WView, 0) * 0.5) + 64) - (_a * 8), UPPER_OFFSET + 3);
+             
+             for (var _a = 0; _a < _stats[1]; _a++)
+                 draw_sprite(sFX_StarRotate, 0, (scrViewX() + (__view_get(e__VW.WView, 0) * 0.5) + 64) - (_a * 8), UPPER_OFFSET + 11);
+             
+             for (var _a = 0; _a < _stats[2]; _a++)
+                 draw_sprite(sFX_StarRotate, 0, (scrViewX() + (__view_get(e__VW.WView, 0) * 0.5) + 64) - (_a * 8), UPPER_OFFSET + 19);
+         }
+     }
+ 
+     if (instance_exists(o22_HandThree))
+     {
+         if (o22_HandThree.active)
+         {
+             _stats[0] = 2;
+             _stats[1] = 2;
+             _stats[2] = 2;
+             
+             switch (o22_HandThree.sel)
+             {
+                 case 0:
+                     str = scrString("name_kotaro");
+                     _stats[2] = 3;
+                     break;
+                 
+                 case 1:
+                     str = scrString("name_ayumi");
+                     _stats[0] = 3;
+                     break;
+                 
+                 case 2:
+                     str = scrString("name_tomoe");
+                     _stats[1] = 3;
+                     break;
+                 
+                 case 3:
+                     str = scrString("name_raizo");
+                     _stats[1] = 1;
+                     _stats[2] = 3;
+                     break;
+                 
+                 case 4:
+                     str = scrString("name_chiyome");
+                     _stats[0] = 3;
+                     _stats[2] = 1;
+                     break;
+                 
+                 case 5:
+                     str = scrString("name_yamada");
+                     _stats[0] = 1;
+                     _stats[1] = 3;
+                     break;
+             }
+             
+             scrSetFont(global.fontTall);
+             str = string_even(str, 3);
+             scrDrawTextCenteredPoint(str, 56, LOWER_OFFSET, 8);
+             scrSetFont(global.fontDefault);
+             
+             for (var _a = 0; _a < _stats[0]; _a++)
+                 draw_sprite_ext(sFX_StarRotate, 0, (scrViewX() + (__view_get(e__VW.WView, 0) * 0.5) - 64) + (_a * 8), LOWER_OFFSET + 4, 1.2, 1, 0, c_black, 1);
+             
+             for (var _a = 0; _a < _stats[1]; _a++)
+                 draw_sprite_ext(sFX_StarRotate, 0, (scrViewX() + (__view_get(e__VW.WView, 0) * 0.5) - 64) + (_a * 8), LOWER_OFFSET + 12, 1.2, 1, 0, c_black, 1);
+             
+             for (var _a = 0; _a < _stats[2]; _a++)
+                 draw_sprite_ext(sFX_StarRotate, 0, (scrViewX() + (__view_get(e__VW.WView, 0) * 0.5) - 64) + (_a * 8), LOWER_OFFSET + 20, 1.2, 1, 0, c_black, 1);
+             
+             for (var _a = 0; _a < _stats[0]; _a++)
+                 draw_sprite(sFX_StarRotate, 0, (scrViewX() + (__view_get(e__VW.WView, 0) * 0.5) - 64) + (_a * 8), LOWER_OFFSET + 3);
+             
+             for (var _a = 0; _a < _stats[1]; _a++)
+                 draw_sprite(sFX_StarRotate, 0, (scrViewX() + (__view_get(e__VW.WView, 0) * 0.5) - 64) + (_a * 8), LOWER_OFFSET + 11);
+             
+             for (var _a = 0; _a < _stats[2]; _a++)
+                 draw_sprite(sFX_StarRotate, 0, (scrViewX() + (__view_get(e__VW.WView, 0) * 0.5) - 64) + (_a * 8), LOWER_OFFSET + 19);
+         }
+     }
+ 
+     if (instance_exists(o22_HandFour))
+     {
+         if (o22_HandFour.active)
+         {
+             _stats[0] = 2;
+             _stats[1] = 2;
+             _stats[2] = 2;
+             
+             switch (o22_HandFour.sel)
+             {
+                 case 0:
+                     str = scrString("name_kotaro");
+                     _stats[2] = 3;
+                     break;
+                 
+                 case 1:
+                     str = scrString("name_ayumi");
+                     _stats[0] = 3;
+                     break;
+                 
+                 case 2:
+                     str = scrString("name_tomoe");
+                     _stats[1] = 3;
+                     break;
+                 
+                 case 3:
+                     str = scrString("name_raizo");
+                     _stats[1] = 1;
+                     _stats[2] = 3;
+                     break;
+                 
+                 case 4:
+                     str = scrString("name_chiyome");
+                     _stats[0] = 3;
+                     _stats[2] = 1;
+                     break;
+                 
+                 case 5:
+                     str = scrString("name_yamada");
+                     _stats[0] = 1;
+                     _stats[1] = 3;
+                     break;
+             }
+             
+             scrSetFont(global.fontTall);
+             str = string_even(str, 3);
+             scrDrawTextCenteredPoint(str, 328, LOWER_OFFSET, 8);
+             scrSetFont(global.fontDefault);
+             
+             for (var _a = 0; _a < _stats[0]; _a++)
+                 draw_sprite_ext(sFX_StarRotate, 0, (scrViewX() + (__view_get(e__VW.WView, 0) * 0.5) + 64) - (_a * 8), LOWER_OFFSET + 4, 1.2, 1, 0, c_black, 1);
+             
+             for (var _a = 0; _a < _stats[1]; _a++)
+                 draw_sprite_ext(sFX_StarRotate, 0, (scrViewX() + (__view_get(e__VW.WView, 0) * 0.5) + 64) - (_a * 8), LOWER_OFFSET + 12, 1.2, 1, 0, c_black, 1);
+             
+             for (var _a = 0; _a < _stats[2]; _a++)
+                 draw_sprite_ext(sFX_StarRotate, 0, (scrViewX() + (__view_get(e__VW.WView, 0) * 0.5) + 64) - (_a * 8), LOWER_OFFSET + 20, 1.2, 1, 0, c_black, 1);
+             
+             for (var _a = 0; _a < _stats[0]; _a++)
+                 draw_sprite(sFX_StarRotate, 0, (scrViewX() + (__view_get(e__VW.WView, 0) * 0.5) + 64) - (_a * 8), LOWER_OFFSET + 3);
+             
+             for (var _a = 0; _a < _stats[1]; _a++)
+                 draw_sprite(sFX_StarRotate, 0, (scrViewX() + (__view_get(e__VW.WView, 0) * 0.5) + 64) - (_a * 8), LOWER_OFFSET + 11);
+             
+             for (var _a = 0; _a < _stats[2]; _a++)
+                 draw_sprite(sFX_StarRotate, 0, (scrViewX() + (__view_get(e__VW.WView, 0) * 0.5) + 64) - (_a * 8), LOWER_OFFSET + 19);
+         }
+     }
+     
+     scrSetFont(global.fontDefault);
+     scrStringDrawCenterAltExt(scrViewX() + (__view_get(e__VW.WView, 0) * 0.5), UPPER_OFFSET, "speed", 0, 16, 4);
+     scrStringDrawCenterAltExt(scrViewX() + (__view_get(e__VW.WView, 0) * 0.5), UPPER_OFFSET + 8, "control", 0, 16, 4);
+     scrStringDrawCenterAltExt(scrViewX() + (__view_get(e__VW.WView, 0) * 0.5), UPPER_OFFSET + 16, "power", 0, 16, 4);
+     scrStringDrawCenterAltExt(scrViewX() + (__view_get(e__VW.WView, 0) * 0.5), LOWER_OFFSET, "speed", 0, 16, 4);
+     scrStringDrawCenterAltExt(scrViewX() + (__view_get(e__VW.WView, 0) * 0.5), LOWER_OFFSET + 8, "control", 0, 16, 4);
+     scrStringDrawCenterAltExt(scrViewX() + (__view_get(e__VW.WView, 0) * 0.5), LOWER_OFFSET + 16, "power", 0, 16, 4);
+     
+     draw_text_ce(56, 424, scrStringVal("win_count", string(wins[LT])), 4);
+     draw_text_ce(328, 424, scrStringVal("win_count", string(wins[RT])), 4);
+ }
  else if (state == STATE_MATCH_PREP)
***************
*** 335,337 ****
                  
!                 if (numPlayers == 4)
                      str = scrString("team") + " " + str;
--- 626,628 ----
                  
!                 if (numPlayers == 4 || numPlayers == 5)
                      str = scrString("team") + " " + str;
***************
*** 344,346 ****
                  
!                 if (numPlayers == 4)
                      str = scrString("team") + " " + str;
--- 635,637 ----
                  
!                 if (numPlayers == 4 || numPlayers == 5)
                      str = scrString("team") + " " + str;
***************
*** 389,391 ****
              
!             if (numPlayers == 4)
              {
--- 680,682 ----
              
!             if (numPlayers == 4 || numPlayers == 5)
              {
***************
*** 420,422 ****
              
!             if (numPlayers == 4)
              {
--- 711,713 ----
              
!             if (numPlayers == 4 || numPlayers == 5)
              {
***************
*** 476,478 ****
          
!         if (numPlayers == 4)
          {
--- 767,769 ----
          
!         if (numPlayers == 4 || numPlayers == 5)
          {
