*** original_scripts/bushido_ball/gml_Object_o22_Game_Other_10.gml	Fri May 23 16:06:08 2025
--- ufo50_modded_scripts/bushido_ball/gml_Object_o22_Game_Other_10.gml	Fri May 23 14:18:17 2025
***************
*** 20,22 ****
      }
!     else if (numPlayers == 2 || numPlayers == 4)
      {
--- 20,22 ----
      }
!     else if (numPlayers >= 2)
      {
***************
*** 24,25 ****
--- 24,31 ----
          p2_sel = -1;
+ 
+         if (numPlayers == 5)
+         {
+             p3_sel = -1;
+             p4_sel = -1;
+         }
          
