*** original_scripts/bushido_ball/gml_Object_o22_HandLeft_Step_0.gml	Thu Aug 28 19:59:00 2025
--- ufo50_modded_scripts/bushido_ball/gml_Object_o22_HandLeft_Step_0.gml	Sat Aug 30 16:48:09 2025
***************
*** 1,2 ****
! scrGetInput(0);
  
--- 1,2 ----
! scrGetInput(o22_Game.controllerMap[0]);
  
***************
*** 29,30 ****
--- 29,34 ----
      o22_CharselLeft.val = sel;
+     
+     if (o22_Game.numPlayers == 5)
+         o22_CharselLeft.y = 300;
+     
      o22_PortraitLeft.val = sel;
***************
*** 96,99 ****
              
!             with (o22_PortraitRight)
!                 visible = true;
          }
--- 100,105 ----
              
!             if (o22_Game.numPlayers != 5) {
!                 with (o22_PortraitRight)
!                     visible = true;
!             }
          }
***************
*** 101,110 ****
      
!     if (o22_Game.numPlayers == 2)
      {
!         var _canBackOut = true;
          
!         with (o22_HandRight)
          {
!             if (ready)
!                 _canBackOut = false;
          }
--- 107,115 ----
      
!     if (o22_Game.numPlayers == 2 || o22_Game.numPlayers == 5)
      {
!         var _canBackOut = !o22_HandRight.ready;
          
!         if (o22_Game.numPlayers == 5)
          {
!             _canBackOut = _canBackOut || !o22_HandThree.ready || !o22_HandFour.ready;
          }
