*** original_scripts/bushido_ball/gml_Object_o22_Game_Other_11.gml	Mon May 26 11:39:34 2025
--- ufo50_modded_scripts/bushido_ball/gml_Object_o22_Game_Other_11.gml	Mon May 26 11:19:47 2025
***************
*** 39,41 ****
          
!         if (o22_Game.numPlayers == 4)
          {
--- 39,41 ----
          
!         if (o22_Game.numPlayers == 4 || o22_Game.numPlayers == 5)
          {
***************
*** 71,77 ****
          
!         if (o22_Game.numPlayers == 2)
          {
              ai = false;
          }
!         else if (o22_Game.numPlayers == 4)
          {
--- 71,80 ----
          
!         if (o22_Game.numPlayers == 2 || o22_Game.numPlayers == 5)
          {
              ai = false;
+             if (o22_Game.numPlayers == 5) {
+                 player = 2;
+             }
          }
!         if (o22_Game.numPlayers == 4 || o22_Game.numPlayers == 5)
          {
***************
*** 92,93 ****
--- 95,100 ----
                  y = other.ystart + 32;
+                 if (o22_Game.numPlayers == 5) {
+                     ai = false;
+                     player = 3;
+                 }
              }
***************
*** 96,98 ****
      
!     if (numPlayers == 4)
      {
--- 103,112 ----
      
!     if (numPlayers == 5)
!     {
!         scr22_CharacterTraits(p1_sel, o22_PlayerL);
!         scr22_CharacterTraits(p2_sel, o22_PlayerL2);
!         scr22_CharacterTraits(p3_sel, o22_PlayerR);
!         scr22_CharacterTraits(p4_sel, o22_PlayerR2);
!     }
!     else if (numPlayers == 4)
      {
