*** original_scripts/bushido_ball/gml_Object_o22_Game_Other_11.gml	Fri Jun  6 20:57:21 2025
--- ufo50_modded_scripts/bushido_ball/gml_Object_o22_Game_Other_11.gml	Fri Jun  6 20:32:05 2025
***************
*** 38,41 ****
          aiLevel = 10;
          
!         if (o22_Game.numPlayers == 4)
          {
--- 38,42 ----
          aiLevel = 10;
+         player = o22_Game.controllerMap[0];
          
!         if (o22_Game.numPlayers == 4 || o22_Game.numPlayers == 5)
          {
***************
*** 54,56 ****
                  aiLevelStart = aiLevel;
!                 player = 1;
                  y = other.ystart + 32;
--- 55,57 ----
                  aiLevelStart = aiLevel;
!                 player = o22_Game.controllerMap[1];
                  y = other.ystart + 32;
***************
*** 71,77 ****
          
!         if (o22_Game.numPlayers == 2)
          {
              ai = false;
          }
!         else if (o22_Game.numPlayers == 4)
          {
--- 72,81 ----
          
!         if (o22_Game.numPlayers == 2 || o22_Game.numPlayers == 5)
          {
              ai = false;
+             if (o22_Game.numPlayers == 5) {
+                 player = o22_Game.controllerMap[2];
+             }
          }
!         if (o22_Game.numPlayers == 4 || o22_Game.numPlayers == 5)
          {
***************
*** 92,93 ****
--- 96,101 ----
                  y = other.ystart + 32;
+                 if (o22_Game.numPlayers == 5) {
+                     ai = false;
+                     player = o22_Game.controllerMap[3];
+                 }
              }
***************
*** 96,98 ****
      
!     if (numPlayers == 4)
      {
--- 104,113 ----
      
!     if (numPlayers == 5)
!     {
!         scr22_CharacterTraits(p1_sel, o22_PlayerL);
!         scr22_CharacterTraits(p2_sel, o22_PlayerL2);
!         scr22_CharacterTraits(p3_sel, o22_PlayerR);
!         scr22_CharacterTraits(p4_sel, o22_PlayerR2);
!     }
!     else if (numPlayers == 4)
      {
