*** original_scripts/bushido_ball/gml_GlobalScript_scr22_Init.gml	Fri Aug 22 11:55:08 2025
--- ufo50_modded_scripts/bushido_ball/gml_GlobalScript_scr22_Init.gml	Wed Aug 20 18:59:27 2025
***************
*** 11,12 ****
--- 11,13 ----
      global.g22_weaponTap = scrReadReal("weaponTap", true);
+     global.g22_randomiseControllers = scrReadReal("randomiseControllers", false);
      scrCloseCurrFile();
