--- original_scripts/gml_Object_oTitleScreens_Create_0.gml
+++ modded_scripts\gml_Object_oTitleScreens_Create_0.gml
@@ -1 +1 @@
-game_set_speed(global.STANDARD_FPS, 0);
+game_set_speed(global.STANDARD_FPS, gamespeed_fps);
@@ -13 +13,0 @@
-        {
@@ -15 +14,0 @@
-        }
@@ -17 +15,0 @@
-        {
@@ -19 +17 @@
-        }
+        
@@ -21,0 +20 @@
+    
@@ -32,0 +32 @@
+        
@@ -33,0 +34 @@
+    
@@ -44,0 +46 @@
+        
@@ -45,0 +48 @@
+    
@@ -54,2 +58,4 @@
-newGameStarted = scrReadNewGameStatus();
-newGame2PStarted = scrReadNewGame2PStatus();
+newGameStarted = scrReadReal("newGame", false);
+newGame2PStarted = scrReadReal("newGame2P", false);
+newGame3PStarted = scrReadReal("newGame3P", false);
+newGame4PStarted = scrReadReal("newGame4P", false);
@@ -61 +67,0 @@
-    {
@@ -63 +69 @@
-    }
+    
@@ -80 +86,0 @@
-    {
@@ -82 +88 @@
-    }
+    
@@ -84 +89,0 @@
-    {
@@ -86 +91 @@
-    }
+    
@@ -92 +97,0 @@
-{
@@ -94 +98,0 @@
-}
@@ -96 +99,0 @@
-{
@@ -98 +100,0 @@
-}
@@ -100 +101,0 @@
-{
@@ -102 +102,0 @@
-}
@@ -104 +103,0 @@
-{
@@ -106 +104,0 @@
-}
@@ -108 +105,0 @@
-{
@@ -110 +107 @@
-}
+
