--- original_scripts/gml_GlobalScript_scrSaveGame.gml
+++ modded_scripts\gml_GlobalScript_scrSaveGame.gml
@@ -4 +3,0 @@
-    {
@@ -6 +5 @@
-    }
+    
@@ -8 +6,0 @@
-    {
@@ -10 +8 @@
-    }
+    
@@ -12 +9,0 @@
-    {
@@ -14 +11 @@
-    }
+    
@@ -16,0 +14 @@
+    
@@ -18 +15,0 @@
-    {
@@ -20 +17 @@
-    }
+    
@@ -22 +18,0 @@
-    {
@@ -24 +20 @@
-    }
+    
@@ -28 +23,0 @@
-        {
@@ -30 +25 @@
-        }
+        
@@ -32 +26,0 @@
-        {
@@ -33,0 +28,6 @@
+
+        if (global.numPlayers == 3)
+            scrWriteReal("newGame3P", true);
+        
+        if (global.numPlayers == 4)
+            scrWriteReal("newGame4P", true);
@@ -35 +35 @@
-    }
+    
@@ -36,0 +37 @@
+    
@@ -39,0 +41 @@
+        
@@ -41 +42,0 @@
-        {
@@ -43 +44 @@
-        }
+        
@@ -46,0 +48 @@
+    
@@ -47,0 +50 @@
+    
@@ -56,0 +60 @@
+    
