--- original_scripts/gml_GlobalScript_scrInitInput.gml
+++ modded_scripts\gml_GlobalScript_scrInitInput.gml
@@ -5,0 +6,2 @@
+    global.joySlot[2] = 2;
+    global.joySlot[3] = 3;
@@ -7,0 +10 @@
+    
@@ -9 +11,0 @@
-    {
@@ -11 +13 @@
-    }
+    
@@ -51 +53,30 @@
-    for (var p = 0; p <= 1; p++)
+    global.DEFAULT_KEY[2][global.INP_UP] = 101;
+    global.DEFAULT_KEY[2][global.INP_DOWN] = 98;
+    global.DEFAULT_KEY[2][global.INP_LEFT] = 97;
+    global.DEFAULT_KEY[2][global.INP_RIGHT] = 99;
+    global.DEFAULT_KEY[2][global.INP_FIRE1] = 35;
+    global.DEFAULT_KEY[2][global.INP_FIRE2] = 46;
+    global.DEFAULT_KEY[2][global.INP_START] = 33;
+    global.DEFAULT_JOY[2][global.INP_UP] = 32781;
+    global.DEFAULT_JOY[2][global.INP_DOWN] = 32782;
+    global.DEFAULT_JOY[2][global.INP_LEFT] = 32783;
+    global.DEFAULT_JOY[2][global.INP_RIGHT] = 32784;
+    global.DEFAULT_JOY[2][global.INP_FIRE1] = 32771;
+    global.DEFAULT_JOY[2][global.INP_FIRE2] = 32769;
+    global.DEFAULT_JOY[2][global.INP_START] = 32778;
+    global.DEFAULT_KEY[3][global.INP_UP] = 101;
+    global.DEFAULT_KEY[3][global.INP_DOWN] = 98;
+    global.DEFAULT_KEY[3][global.INP_LEFT] = 97;
+    global.DEFAULT_KEY[3][global.INP_RIGHT] = 99;
+    global.DEFAULT_KEY[3][global.INP_FIRE1] = 35;
+    global.DEFAULT_KEY[3][global.INP_FIRE2] = 46;
+    global.DEFAULT_KEY[3][global.INP_START] = 33;
+    global.DEFAULT_JOY[3][global.INP_UP] = 32781;
+    global.DEFAULT_JOY[3][global.INP_DOWN] = 32782;
+    global.DEFAULT_JOY[3][global.INP_LEFT] = 32783;
+    global.DEFAULT_JOY[3][global.INP_RIGHT] = 32784;
+    global.DEFAULT_JOY[3][global.INP_FIRE1] = 32771;
+    global.DEFAULT_JOY[3][global.INP_FIRE2] = 32769;
+    global.DEFAULT_JOY[3][global.INP_START] = 32778;
+    
+    for (var p = 0; p <= 3; p++)
@@ -54 +84,0 @@
-        {
@@ -56 +86 @@
-        }
+        
@@ -58 +87,0 @@
-        {
@@ -61 +90 @@
-    }
+    
@@ -63,0 +93,2 @@
+    global.doubledUpButtons[2] = scrReadConfig("doubledUpButtonsP3", true);
+    global.doubledUpButtons[3] = scrReadConfig("doubledUpButtonsP4", true);
@@ -65,0 +97 @@
+    
@@ -73,0 +106 @@
+            
@@ -77,0 +111 @@
+            
@@ -81,0 +116 @@
+            
@@ -85,0 +121 @@
+            
@@ -89,0 +126 @@
+            
@@ -93,0 +131 @@
+            
@@ -97,0 +136 @@
+            
@@ -101,0 +141 @@
+            
@@ -105,0 +146 @@
+            
@@ -109,0 +151 @@
+            
@@ -113,0 +156 @@
+            
@@ -117,0 +161 @@
+            
@@ -124,0 +169 @@
+            
@@ -128,0 +174 @@
+            
@@ -132,0 +179 @@
+            
@@ -145,0 +193 @@
+            
@@ -174,0 +223 @@
+            
@@ -187,0 +237 @@
+            
@@ -191,0 +242 @@
+            
@@ -195,0 +247 @@
+            
@@ -199,0 +252 @@
+            
@@ -203,0 +257 @@
+            
@@ -207,0 +262 @@
+            
@@ -211,0 +267 @@
+            
@@ -225,0 +282 @@
+            
@@ -229,0 +287 @@
+            
@@ -233,0 +292 @@
+            
@@ -238,0 +298 @@
+            
@@ -243,0 +304 @@
+            
@@ -247,0 +309 @@
+            
@@ -251,0 +314 @@
+            
@@ -255,0 +319 @@
+            
@@ -259,0 +324 @@
+            
@@ -263,0 +329 @@
+            
@@ -267,0 +334 @@
+            
@@ -271,0 +339 @@
+            
@@ -275,0 +344 @@
+            
@@ -279,0 +349 @@
+            
@@ -283,0 +354 @@
+            
@@ -287,0 +359 @@
+            
@@ -293,0 +366 @@
+    
@@ -308,0 +382,3 @@
+    global.joyType[2] = scrReadConfig("joyTypeP3", scrDetectJoyType(2));
+    global.joyType[3] = scrReadConfig("joyTypeP4", scrDetectJoyType(3));
+    
@@ -317,0 +394 @@
+            
@@ -322,0 +400 @@
+            
@@ -327,0 +406 @@
+            
@@ -332,0 +412 @@
+            
@@ -337,0 +418 @@
+            
@@ -342,0 +424 @@
+            
@@ -347,0 +430 @@
+            
@@ -352,0 +436 @@
+            
@@ -357,0 +442 @@
+            
@@ -362,0 +448 @@
+            
@@ -367,0 +454 @@
+            
@@ -372,0 +460 @@
+            
@@ -377,0 +466 @@
+            
@@ -382,0 +472 @@
+            
@@ -388,0 +479 @@
+    
