--- original_scripts/gml_Object_oPauseMenu_Other_13.gml
+++ modded_scripts\gml_Object_oPauseMenu_Other_13.gml
@@ -10 +11,2 @@
-    for (var p = 0; p <= 1; p++)
+    
+    for (var p = 0; p <= 3; p++)
@@ -13 +14,0 @@
-        {
@@ -16 +17 @@
-    }
+    
@@ -25,0 +27 @@
+    
@@ -33,0 +36 @@
+    
@@ -36,0 +40,2 @@
+    gamepadConnect[2] = gamepad_is_connected(global.joySlot[2]);
+    gamepadConnect[3] = gamepad_is_connected(global.joySlot[3]);
@@ -39,0 +45 @@
+    
@@ -71,0 +78 @@
+    
@@ -84,0 +92 @@
+    
@@ -85,0 +94 @@
+    
@@ -91,0 +101 @@
+    
@@ -97 +107,2 @@
-    if (keyboard_check_pressed(112))
+    
+    if (keyboard_check_pressed(vk_f1))
@@ -103,2 +114,2 @@
-    if (gamepad_button_check(joySlot, 32773) && gamepad_button_check(joySlot, 32774))
-    {
+    
+    if (gamepad_button_check(joySlot, gp_shoulderl) && gamepad_button_check(joySlot, gp_shoulderr))
@@ -106 +116,0 @@
-    }
@@ -108 +117,0 @@
-    {
@@ -110 +119 @@
-    }
+    
@@ -117,0 +127 @@
+    
@@ -126,0 +137 @@
+            
@@ -131,0 +143 @@
+            
@@ -136,0 +149 @@
+            
@@ -147,0 +161 @@
+            
@@ -159,2 +173 @@
-        if (keyboard_check_released(global.keyMap[0][global.INP_FIRE2]) || keyboard_check_released(global.keyMap[1][global.INP_FIRE2]) || gamepad_button_check_released(global.joySlot[0], global.joyMap[0][global.INP_FIRE2]) || gamepad_button_check_released(global.joySlot[1], global.joyMap[1][global.INP_FIRE2]) || keyboard_check_released(13))
-        {
+        if (keyboard_check_released(global.keyMap[0][global.INP_FIRE2]) || keyboard_check_released(global.keyMap[1][global.INP_FIRE2]) || gamepad_button_check_released(global.joySlot[0], global.joyMap[0][global.INP_FIRE2]) || gamepad_button_check_released(global.joySlot[1], global.joyMap[1][global.INP_FIRE2]) || keyboard_check_released(vk_enter))
@@ -163 +175,0 @@
-    }
@@ -166 +178 @@
-        if (keyboard_check_pressed(1))
+        if (keyboard_check_pressed(vk_anykey))
@@ -170,0 +183 @@
+        
@@ -175,0 +189 @@
+        
@@ -176,0 +191 @@
+        
@@ -179,0 +195 @@
+            
@@ -181 +196,0 @@
-            {
@@ -183 +198 @@
-            }
+            
@@ -185 +199,0 @@
-            {
@@ -187 +201 @@
-            }
+            
@@ -189 +202,0 @@
-            {
@@ -191 +204 @@
-            }
+            
@@ -193 +205,0 @@
-            {
@@ -195 +207 @@
-            }
+            
@@ -197 +208,0 @@
-            {
@@ -199 +210 @@
-            }
+            
@@ -201 +211,0 @@
-            {
@@ -203 +213 @@
-            }
+            
@@ -205 +214,0 @@
-            {
@@ -207 +216 @@
-            }
+            
@@ -208,0 +218 @@
+            
@@ -210 +219,0 @@
-            {
@@ -212 +220,0 @@
-            }
@@ -214 +221,0 @@
-            {
@@ -216 +223 @@
-            }
+            
@@ -219,0 +227 @@
+                
@@ -225 +232,0 @@
-                        {
@@ -229 +236 @@
-                }
+                
@@ -230,0 +238 @@
+                
@@ -234 +241,0 @@
-                    {
@@ -237 +244 @@
-                }
+                
@@ -257 +264 @@
-        for (var p = 0; p <= 1; p++)
+        for (var p = 0; p <= 3; p++)
@@ -260 +266,0 @@
-            {
@@ -263,2 +269,2 @@
-        }
-        for (var p = 0; p <= 1; p++)
+        
+        for (var p = 0; p <= 3; p++)
@@ -267 +272,0 @@
-            {
@@ -270 +275 @@
-        }
+        
@@ -274,0 +280,2 @@
+        global.joyType[2] = scrDetectJoyType(2);
+        global.joyType[3] = scrDetectJoyType(3);
@@ -276,0 +284 @@
+    
@@ -279,0 +288 @@
+        
@@ -281 +289,0 @@
-        {
@@ -285 +292,0 @@
-}
@@ -290,2 +297 @@
-        if (keyboard_check_released(global.keyMap[0][global.INP_FIRE2]) || keyboard_check_released(global.keyMap[1][global.INP_FIRE2]) || gamepad_button_check_released(global.joySlot[0], global.joyMap[0][global.INP_FIRE2]) || gamepad_button_check_released(global.joySlot[1], global.joyMap[1][global.INP_FIRE2]) || keyboard_check_released(13))
-        {
+        if (keyboard_check_released(global.keyMap[0][global.INP_FIRE2]) || keyboard_check_released(global.keyMap[1][global.INP_FIRE2]) || gamepad_button_check_released(global.joySlot[0], global.joyMap[0][global.INP_FIRE2]) || gamepad_button_check_released(global.joySlot[1], global.joyMap[1][global.INP_FIRE2]) || keyboard_check_released(vk_enter))
@@ -294 +299,0 @@
-    }
@@ -297 +302 @@
-        if (keyboard_check_pressed(1))
+        if (keyboard_check_pressed(vk_anykey))
@@ -302,0 +308 @@
+        
@@ -303,0 +310 @@
+        
@@ -307,2 +314,2 @@
-            if (gamepad_button_check_pressed(_slot, 32769))
-            {
+            
+            if (gamepad_button_check_pressed(_slot, gp_face1))
@@ -310,3 +317,2 @@
-            }
-            if (gamepad_button_check_pressed(_slot, 32770))
-            {
+            
+            if (gamepad_button_check_pressed(_slot, gp_face2))
@@ -314,3 +320,2 @@
-            }
-            if (gamepad_button_check_pressed(_slot, 32771))
-            {
+            
+            if (gamepad_button_check_pressed(_slot, gp_face3))
@@ -318,3 +323,2 @@
-            }
-            if (gamepad_button_check_pressed(_slot, 32772))
-            {
+            
+            if (gamepad_button_check_pressed(_slot, gp_face4))
@@ -322,3 +326,2 @@
-            }
-            if (gamepad_button_check_pressed(_slot, 32773))
-            {
+            
+            if (gamepad_button_check_pressed(_slot, gp_shoulderl))
@@ -326,3 +329,2 @@
-            }
-            if (gamepad_button_check_pressed(_slot, 32775))
-            {
+            
+            if (gamepad_button_check_pressed(_slot, gp_shoulderlb))
@@ -330,3 +332,2 @@
-            }
-            if (gamepad_button_check_pressed(_slot, 32774))
-            {
+            
+            if (gamepad_button_check_pressed(_slot, gp_shoulderr))
@@ -334,3 +335,2 @@
-            }
-            if (gamepad_button_check_pressed(_slot, 32776))
-            {
+            
+            if (gamepad_button_check_pressed(_slot, gp_shoulderrb))
@@ -338,3 +338,2 @@
-            }
-            if (gamepad_button_check_pressed(_slot, 32781))
-            {
+            
+            if (gamepad_button_check_pressed(_slot, gp_padu))
@@ -342,3 +341,2 @@
-            }
-            if (gamepad_button_check_pressed(_slot, 32782))
-            {
+            
+            if (gamepad_button_check_pressed(_slot, gp_padd))
@@ -346,3 +344,2 @@
-            }
-            if (gamepad_button_check_pressed(_slot, 32783))
-            {
+            
+            if (gamepad_button_check_pressed(_slot, gp_padl))
@@ -350,3 +347,2 @@
-            }
-            if (gamepad_button_check_pressed(_slot, 32784))
-            {
+            
+            if (gamepad_button_check_pressed(_slot, gp_padr))
@@ -355 +351 @@
-        }
+        
@@ -360,0 +357 @@
+            
@@ -362 +358,0 @@
-            {
@@ -364 +360 @@
-            }
+            
@@ -369,0 +366 @@
+            
