--- original_scripts/gml_GlobalScript_scrInputClear.gml
+++ modded_scripts\gml_GlobalScript_scrInputClear.gml
@@ -21,0 +22 @@
+    
@@ -24 +25 @@
-        for (var p = 0; p <= 1; p++)
+        for (var p = 0; p <= 3; p++)
@@ -30,0 +32 @@
+    
@@ -33,5 +35,5 @@
-        keyboard_clear(13);
-        keyboard_clear(38);
-        keyboard_clear(40);
-        keyboard_clear(37);
-        keyboard_clear(39);
+        keyboard_clear(vk_enter);
+        keyboard_clear(vk_up);
+        keyboard_clear(vk_down);
+        keyboard_clear(vk_left);
+        keyboard_clear(vk_right);
