--- original_scripts/gml_GlobalScript_scrMenuNavigation.gml
+++ modded_scripts\gml_GlobalScript_scrMenuNavigation.gml
@@ -4 +3,0 @@
-    {
@@ -6 +5 @@
-    }
+    
@@ -8 +6,0 @@
-    {
@@ -10 +8 @@
-    }
+    
@@ -12 +10,2 @@
-    if (keyboard_check_pressed(38))
+    
+    if (keyboard_check_pressed(vk_up))
@@ -16,0 +16 @@
+            
@@ -18 +17,0 @@
-            {
@@ -21 +19,0 @@
-        }
@@ -24 +22 @@
-    else if (keyboard_check_pressed(40))
+    else if (keyboard_check_pressed(vk_down))
@@ -28,0 +27 @@
+            
@@ -30 +28,0 @@
-            {
@@ -33 +30,0 @@
-        }
@@ -40,0 +38 @@
+            
@@ -42 +39,0 @@
-            {
@@ -45 +41,0 @@
-        }
@@ -52,0 +49 @@
+            
@@ -54 +50,0 @@
-            {
@@ -57 +52,0 @@
-        }
@@ -59,0 +55 @@
+    
@@ -65,0 +62 @@
+    
@@ -77,0 +75 @@
+    
@@ -83 +81,2 @@
-        if (keyboard_check_pressed(37) && _numItems > 1)
+        
+        if (keyboard_check_pressed(vk_left) && _numItems > 1)
@@ -93 +92 @@
-        else if (keyboard_check_pressed(13))
+        else if (keyboard_check_pressed(vk_enter))
@@ -95,0 +95 @@
+            
@@ -99 +98,0 @@
-                {
@@ -101 +99,0 @@
-                }
@@ -103 +100,0 @@
-                {
@@ -106 +103 @@
-            }
+            
@@ -109 +106 @@
-        else if (keyboard_check_pressed(39))
+        else if (keyboard_check_pressed(vk_right))
@@ -117 +113,0 @@
-            {
@@ -119 +115 @@
-            }
+            
@@ -121,0 +118 @@
+        
@@ -124,0 +122 @@
+            
@@ -128 +125,0 @@
-                {
@@ -130 +126,0 @@
-                }
@@ -132 +127,0 @@
-                {
@@ -135 +130 @@
-            }
+            
@@ -141,0 +137 @@
+            
@@ -145 +140,0 @@
-                {
@@ -147 +141,0 @@
-                }
@@ -149 +142,0 @@
-                {
@@ -152 +145 @@
-            }
+            
@@ -155,0 +149 @@
+    
@@ -158 +152 @@
-        if (keyboard_check_pressed(13))
+        if (keyboard_check_pressed(vk_enter))
@@ -160,0 +155 @@
+            
@@ -164 +158,0 @@
-                {
@@ -166 +159,0 @@
-                }
@@ -168 +160,0 @@
-                {
@@ -171 +163 @@
-            }
+            
@@ -177 +168,0 @@
-            {
@@ -179 +170 @@
-            }
+            
@@ -180,0 +172 @@
+            
@@ -184 +175,0 @@
-                {
@@ -186 +176,0 @@
-                }
@@ -188 +177,0 @@
-                {
@@ -191 +180 @@
-            }
+            
@@ -197 +186 @@
-        if (keyboard_check_pressed(37) || keyboard_check_pressed(39) || keyboard_check_pressed(13))
+        if (keyboard_check_pressed(vk_enter))
@@ -199,2 +188 @@
-            if (keyboard_check_pressed(37))
-            {
+            if (keyboard_check_pressed(vk_left))
@@ -202,3 +190,2 @@
-            }
-            if (keyboard_check_pressed(39))
-            {
+            
+            if (keyboard_check_pressed(vk_right))
@@ -206 +193 @@
-            }
+            
@@ -210 +197 @@
-        else if (pressLeft || pressRight || fire2pressed)
+        else if (pressLeft || keyboard_check_pressed(vk_left))
@@ -212,2 +198,0 @@
-            if (pressLeft)
-            {
@@ -214,0 +200,3 @@
+            
+            itemIndex[menuSel] = max(itemIndex[menuSel] - 1, 0);
+            return itemIndex[menuSel];
@@ -216 +204 @@
-            if (pressRight)
+        else if (pressRight || keyboard_check_pressed(vk_right))
@@ -219,2 +207,2 @@
-            }
-            itemIndex[menuSel] = !itemIndex[menuSel];
+            
+            itemIndex[menuSel] = min(itemIndex[menuSel] + 1, 3);
@@ -226,2 +214 @@
-        if (keyboard_check_pressed(13))
-        {
+        if (keyboard_check_pressed(vk_enter))
@@ -229 +215,0 @@
-        }
@@ -231 +216,0 @@
-        {
@@ -234 +218,0 @@
-    }
@@ -239 +223 @@
-            if (keyboard_check_pressed(13))
+            if (keyboard_check_pressed(vk_enter))
@@ -243,0 +228 @@
+            
@@ -250 +235 @@
-        else if (keyboard_check_pressed(13))
+        else if (keyboard_check_pressed(vk_enter))
@@ -262,0 +248 @@
+    
