*** original_scripts/gml_Object_oPauseMenu_Other_12.gml	Sun May 11 20:13:18 2025
--- ufo50_modded_scripts/gml_Object_oPauseMenu_Other_12.gml	Fri May  9 20:16:24 2025
***************
*** 10,12 ****
      
!     for (var p = 0; p <= 1; p++)
      {
--- 10,12 ----
      
!     for (var p = 0; p <= 3; p++)
      {
***************
*** 141,143 ****
                  
!                 for (var p = 0; p <= 1; p++)
                  {
--- 141,143 ----
                  
!                 for (var p = 0; p <= 3; p++)
                  {
***************
*** 155,157 ****
                  
!                 for (var p = 0; p <= 1; p++)
                  {
--- 155,157 ----
                  
!                 for (var p = 0; p <= 3; p++)
                  {
***************
*** 179,181 ****
      {
!         for (var p = 0; p <= 1; p++)
          {
--- 179,181 ----
      {
!         for (var p = 0; p <= 3; p++)
          {
***************
*** 185,187 ****
          
!         for (var p = 0; p <= 1; p++)
          {
--- 185,187 ----
          
!         for (var p = 0; p <= 3; p++)
          {
