*** original_scripts/gml_Object_oTitleScreens_Other_12.gml	Sun May 11 20:25:31 2025
--- ufo50_modded_scripts/gml_Object_oTitleScreens_Other_12.gml	Sun May 11 20:14:00 2025
***************
*** 59,61 ****
          menuOption[1] = scrStringManual("title_2players", 0);
!         menuOption[2] = scrStringManual("title_high_scores", 0);
      }
--- 59,62 ----
          menuOption[1] = scrStringManual("title_2players", 0);
!         menuOption[2] = "4 PLAYER TEAMS";
!         menuOption[3] = scrStringManual("title_high_scores", 0);
      }
***************
*** 149,152 ****
          menuOption[1] = scrStringManual("title_2p_continue", 0);
!         menuOption[2] = scrStringManual("title_1p_new_game", 0);
!         menuOption[3] = scrStringManual("title_2p_new_game", 0);
          
--- 150,162 ----
          menuOption[1] = scrStringManual("title_2p_continue", 0);
!         menuOption[2] = "3P CONTINUE";
!         menuOption[3] = "4P CONTINUE";
!         menuOption[4] = scrStringManual("title_1p_new_game", 0);
!         menuOption[5] = scrStringManual("title_2p_new_game", 0);
!         menuOption[6] = "3P NEW GAME";
!         menuOption[7] = "4P NEW GAME";
!         
!         if (newGameStarted)
!             menuSel = 0;
!         else
!             menuSel = 4;
          
***************
*** 158,163 ****
          
!         if (newGameStarted)
!             menuSel = 0;
!         else
!             menuSel = 2;
      }
--- 168,174 ----
          
!         if (!newGame3PStarted)
!             menuAvail[2] = false;
!         
!         if (!newGame4PStarted)
!             menuAvail[3] = false;
      }
***************
*** 293,297 ****
                  scrSwitchSub(SUB_CONFIRM);
!             else if (newGameStarted && menuStyle == 15 && menuSel == 2)
                  scrSwitchSub(SUB_CONFIRM);
!             else if (newGame2PStarted && menuStyle == 15 && menuSel == 3)
                  scrSwitchSub(SUB_CONFIRM);
--- 304,312 ----
                  scrSwitchSub(SUB_CONFIRM);
!             else if (newGameStarted && menuStyle == 15 && menuSel == 4)
                  scrSwitchSub(SUB_CONFIRM);
!             else if (newGame2PStarted && menuStyle == 15 && menuSel == 5)
!                 scrSwitchSub(SUB_CONFIRM);
!             else if (newGame3PStarted && menuStyle == 15 && menuSel == 6)
!                 scrSwitchSub(SUB_CONFIRM);
!             else if (newGame4PStarted && menuStyle == 15 && menuSel == 7)
                  scrSwitchSub(SUB_CONFIRM);
***************
*** 299,301 ****
                  scrSwitchState(STATE_HIGHSCORE);
!             else if (menuStyle == 4 && menuSel == 2)
                  scrSwitchState(STATE_HIGHSCORE);
--- 314,316 ----
                  scrSwitchState(STATE_HIGHSCORE);
!             else if (menuStyle == 4 && menuSel == 3)
                  scrSwitchState(STATE_HIGHSCORE);
***************
*** 379,380 ****
--- 394,398 ----
          
+         if (menuStyle == 4 && menuSel == 2)
+             global.numPlayers = 4;
+         
          if (menuStyle == 5 && menuSel == 1)
***************
*** 406,407 ****
--- 424,431 ----
          
+         if (menuStyle == 16 && menuSel == 1)
+             global.numPlayers = 2;
+         
+         if (menuStyle == 17 && menuSel == 1)
+             global.numPlayers = 4;
+ 
          if (menuStyle == 15 && menuSel == 1)
***************
*** 409,417 ****
          
          if (menuStyle == 15 && menuSel == 3)
!             global.numPlayers = 2;
          
!         if (menuStyle == 16 && menuSel == 1)
              global.numPlayers = 2;
          
!         if (menuStyle == 17 && menuSel == 1)
              global.numPlayers = 4;
--- 433,447 ----
          
+         if (menuStyle == 15 && menuSel == 2)
+             global.numPlayers = 3;
+ 
          if (menuStyle == 15 && menuSel == 3)
!             global.numPlayers = 4;
          
!         if (menuStyle == 15 && menuSel == 5)
              global.numPlayers = 2;
          
!         if (menuStyle == 15 && menuSel == 6)
!             global.numPlayers = 3;
!         
!         if (menuStyle == 15 && menuSel == 7)
              global.numPlayers = 4;
