*** original_scripts/gml_Object_oTitleScreens_Other_12.gml	Tue May 27 20:29:36 2025
--- ufo50_modded_scripts/gml_Object_oTitleScreens_Other_12.gml	Tue May 27 19:32:57 2025
***************
*** 72,74 ****
          menuOption[2] = scrStringManual("title_coop", 0);
!         menuOption[3] = scrStringManual("title_options", 0);
      }
--- 72,75 ----
          menuOption[2] = scrStringManual("title_coop", 0);
!         menuOption[3] = "4 PLAYER VS"
!         menuOption[4] = scrStringManual("title_options", 0);
      }
***************
*** 78,79 ****
--- 79,82 ----
          menuOption[1] = scrStringManual("title_2p_battle", 0);
+         menuOption[2] = "3P BATTLE";
+         menuOption[3] = "4P BATTLE";
      }
***************
*** 149,152 ****
          menuOption[1] = scrStringManual("title_2p_continue", 0);
!         menuOption[2] = scrStringManual("title_1p_new_game", 0);
!         menuOption[3] = scrStringManual("title_2p_new_game", 0);
          
--- 152,164 ----
          menuOption[1] = scrStringManual("title_2p_continue", 0);
!         menuOption[2] = "3P CONTINUE";
!         menuOption[3] = "4P CONTINUE";
!         menuOption[4] = scrStringManual("title_1p_new_game", 0);
!         menuOption[5] = scrStringManual("title_2p_new_game", 0);
!         menuOption[6] = "3P NEW GAME";
!         menuOption[7] = "4P NEW GAME";
!         
!         if (newGameStarted)
!             menuSel = 0;
!         else
!             menuSel = 4;
          
***************
*** 158,163 ****
          
!         if (newGameStarted)
!             menuSel = 0;
!         else
!             menuSel = 2;
      }
--- 170,176 ----
          
!         if (!newGame3PStarted)
!             menuAvail[2] = false;
!         
!         if (!newGame4PStarted)
!             menuAvail[3] = false;
      }
***************
*** 174,175 ****
--- 187,210 ----
      }
+     else if (menuStyle == 18)
+     {
+         menuOption[0] = scrStringManual("title_1player", 0);
+         menuOption[1] = scrStringManual("title_2players", 0);
+         menuOption[2] = "4 PLAYER TEAMS";
+         menuOption[3] = scrStringManual("title_high_scores", 0);
+     }
+     else if (menuStyle == 19)
+     {
+         menuOption[0] = scrStringManual("title_1player", 0);
+         menuOption[1] = scrStringManual("title_2players", 0);
+         menuOption[2] = "3 PLAYERS";
+         menuOption[3] = "4 PLAYERS";
+         menuOption[4] = scrStringManual("title_high_scores", 0);
+     }
+     else if (menuStyle == 20)
+     {
+         menuOption[0] = scrStringManual("title_1player", 0);
+         menuOption[1] = scrStringManual("title_2players", 0);
+         menuOption[2] = "3 PLAYERS";
+         menuOption[3] = "4 PLAYERS";
+     }
      else
***************
*** 293,297 ****
                  scrSwitchSub(SUB_CONFIRM);
!             else if (newGameStarted && menuStyle == 15 && menuSel == 2)
                  scrSwitchSub(SUB_CONFIRM);
!             else if (newGame2PStarted && menuStyle == 15 && menuSel == 3)
                  scrSwitchSub(SUB_CONFIRM);
--- 328,336 ----
                  scrSwitchSub(SUB_CONFIRM);
!             else if (newGameStarted && menuStyle == 15 && menuSel == 4)
                  scrSwitchSub(SUB_CONFIRM);
!             else if (newGame2PStarted && menuStyle == 15 && menuSel == 5)
!                 scrSwitchSub(SUB_CONFIRM);
!             else if (newGame3PStarted && menuStyle == 15 && menuSel == 6)
!                 scrSwitchSub(SUB_CONFIRM);
!             else if (newGame4PStarted && menuStyle == 15 && menuSel == 7)
                  scrSwitchSub(SUB_CONFIRM);
***************
*** 301,302 ****
--- 340,345 ----
                  scrSwitchState(STATE_HIGHSCORE);
+             else if (menuStyle == 18 && menuSel == 3)
+                 scrSwitchState(STATE_HIGHSCORE);
+             else if (menuStyle == 19 && menuSel == 4)
+                 scrSwitchState(STATE_HIGHSCORE);
              else if (menuStyle == 10 && menuSel == 3)
***************
*** 378,379 ****
--- 421,446 ----
              global.numPlayers = 2;
+ 
+         if (menuStyle == 18 && menuSel == 1)
+             global.numPlayers = 2;
+         
+         if (menuStyle == 18 && menuSel == 2)
+             global.numPlayers = 4;
+ 
+         if (menuStyle == 19 && menuSel == 1)
+             global.numPlayers = 2;
+         
+         if (menuStyle == 19 && menuSel == 2)
+             global.numPlayers = 3;
+ 
+         if (menuStyle == 19 && menuSel == 3)
+             global.numPlayers = 4;
+ 
+         if (menuStyle == 20 && menuSel == 1)
+             global.numPlayers = 2;
+         
+         if (menuStyle == 20 && menuSel == 2)
+             global.numPlayers = 3;
+ 
+         if (menuStyle == 20 && menuSel == 3)
+             global.numPlayers = 4;
          
***************
*** 387,388 ****
--- 454,461 ----
              global.numPlayers = 2;
+ 
+         if (menuStyle == 7 && menuSel == 2)
+             global.numPlayers = 3;
+ 
+         if (menuStyle == 7 && menuSel == 3)
+             global.numPlayers = 4;
          
***************
*** 406,407 ****
--- 479,486 ----
          
+         if (menuStyle == 16 && menuSel == 1)
+             global.numPlayers = 2;
+         
+         if (menuStyle == 17 && menuSel == 1)
+             global.numPlayers = 4;
+ 
          if (menuStyle == 15 && menuSel == 1)
***************
*** 409,417 ****
          
          if (menuStyle == 15 && menuSel == 3)
!             global.numPlayers = 2;
          
!         if (menuStyle == 16 && menuSel == 1)
              global.numPlayers = 2;
          
!         if (menuStyle == 17 && menuSel == 1)
              global.numPlayers = 4;
--- 488,502 ----
          
+         if (menuStyle == 15 && menuSel == 2)
+             global.numPlayers = 3;
+ 
          if (menuStyle == 15 && menuSel == 3)
!             global.numPlayers = 4;
          
!         if (menuStyle == 15 && menuSel == 5)
              global.numPlayers = 2;
          
!         if (menuStyle == 15 && menuSel == 6)
!             global.numPlayers = 3;
!         
!         if (menuStyle == 15 && menuSel == 7)
              global.numPlayers = 4;
***************
*** 424,425 ****
--- 509,513 ----
          
+         if (menuStyle == 6 && menuSel == 3)
+             global.numPlayers = 5;
+         
          if (menuStyle == 17 && menuSel == 2)
***************
*** 427,429 ****
          
!         if (menuStyle == 6 && menuSel == 3)
              global.numPlayers = 0;
--- 515,517 ----
          
!         if (menuStyle == 6 && menuSel == 4)
              global.numPlayers = 0;
