*** original_scripts/gml_GlobalScript_scrDefineMeta.gml	Sun Dec  7 21:04:41 2025
--- ufo50_modded_scripts/gml_GlobalScript_scrDefineMeta.gml	Sat Dec  6 19:19:35 2025
***************
*** 14,19 ****
          global.mGameRealtime[i] = true;
-         
          for (var j = 0; j < global.NUM_PILLARS; j++)
              global.mGamePillars[i][j] = false;
!         
          global.mGameMenuStyle[i] = 0;
--- 14,19 ----
          global.mGameRealtime[i] = true;
          for (var j = 0; j < global.NUM_PILLARS; j++)
+         {
              global.mGamePillars[i][j] = false;
!         }
          global.mGameMenuStyle[i] = 0;
***************
*** 31,33 ****
      }
-     
      nGame = 1;
--- 31,32 ----
***************
*** 41,48 ****
      scr34_Meta(nGame);
      scrAddPillars(global.PILLAR_QUICK, global.PILLAR_REFLEX);
      global.mGameCoop[nGame] = true;
      nGame++;
      scr49_Meta(nGame);
      scrAddPillars(global.PILLAR_QUICK, global.PILLAR_REFLEX);
!     global.mGameVersus[nGame] = true;
      nGame++;
--- 40,51 ----
      scr34_Meta(nGame);
+     global.mGameMenuStyle[nGame] = 19;
      scrAddPillars(global.PILLAR_QUICK, global.PILLAR_REFLEX);
      global.mGameCoop[nGame] = true;
+     global.mGamePlayersSupported[nGame] = 4;
      nGame++;
      scr49_Meta(nGame);
+     global.mGameMenuStyle[nGame] = 18;
      scrAddPillars(global.PILLAR_QUICK, global.PILLAR_REFLEX);
!     global.mGameVersus[nGame] = true;
!     global.mGamePlayersSupported[nGame] = 4;
      nGame++;
***************
*** 54,55 ****
--- 57,59 ----
      global.mGameCoop[nGame] = true;
+     global.mGamePlayersSupported[nGame] = 4;
      nGame++;
***************
*** 57,58 ****
--- 61,75 ----
      scrAddPillars(global.PILLAR_QUICK, global.PILLAR_REFLEX);
+     global.mGameCoop[nGame] = true;
+     global.mGameMultiplayer[nGame] = true;
+     global.mGameMenuStyle[nGame] = 22;
+     global.mGamePlayersSupported[nGame] = 4;
+     global.g15_threshY = 96;
+     global.g15_skipOptionsMenu = false;
+     global.g15_gameOverCondition = 0;
+     global.g15_playerScores = [0, 0, 0, 0];
+     global.g15_coopCostMultiplier = 1.2;
+     global.g15_isVsMode = false;
+     global.g15_multiplayerStartingLevel = 1;
+     global.g15_multiplayerStartingPowerups = [0, 0, 0, 0, 0, 0];
+     global.g15_randomiseLevel = false;
      nGame++;
***************
*** 64,65 ****
--- 81,84 ----
      global.mGameVersus[nGame] = true;
+     global.mGameMenuStyle[nGame] = 23;
+     global.mGamePlayersSupported[nGame] = 4;
      nGame++;
***************
*** 78,79 ****
--- 97,99 ----
      global.mGameCoop[nGame] = true;
+     global.mGamePlayersSupported[nGame] = 4;
      nGame++;
***************
*** 83,84 ****
--- 103,105 ----
      global.mGameCoop[nGame] = true;
+     global.mGamePlayersSupported[nGame] = 4;
      nGame++;
***************
*** 97,100 ****
--- 118,123 ----
      scr28_Meta(nGame);
+     global.mGameMenuStyle[nGame] = 20;
      scrAddPillars(global.PILLAR_QUICK, global.PILLAR_REFLEX);
      global.mGameVersus[nGame] = true;
+     global.mGamePlayersSupported[nGame] = 6;
      nGame++;
***************
*** 106,107 ****
--- 129,131 ----
      global.mGameVersus[nGame] = true;
+     global.mGamePlayersSupported[nGame] = 4;
      nGame++;
***************
*** 111,112 ****
--- 135,138 ----
      scr32_Meta(nGame);
+     global.mGameMenuStyle[nGame] = 25;
+     global.mGamePlayersSupported[nGame] = 4;
      scrAddPillars(global.PILLAR_QUICK, global.PILLAR_REFLEX);
***************
*** 118,123 ****
--- 144,152 ----
      scr36_Meta(nGame);
+     global.mGameMenuStyle[nGame] = 24;
      scrAddPillars(global.PILLAR_EPIC, global.PILLAR_THINKY);
      global.mGameVersus[nGame] = true;
+     global.mGamePlayersSupported[nGame] = 4;
      nGame++;
      scr43_Meta(nGame);
+     global.mGameMenuStyle[nGame] = 21;
      scrAddPillars(global.PILLAR_QUICK, global.PILLAR_REFLEX);
***************
*** 125,126 ****
--- 154,156 ----
      global.mGameCoop[nGame] = true;
+     global.mGamePlayersSupported[nGame] = 4;
      nGame++;
