*** original_scripts/gml_GlobalScript_scrMenuNavigation.gml	Sat Jun  7 21:06:07 2025
--- ufo50_modded_scripts/gml_GlobalScript_scrMenuNavigation.gml	Sun Jun  8 14:00:58 2025
***************
*** 185,187 ****
      {
!         if (keyboard_check_pressed(vk_left) || keyboard_check_pressed(vk_right) || keyboard_check_pressed(vk_enter))
          {
--- 185,187 ----
      {
!         if (keyboard_check_pressed(vk_enter))
          {
***************
*** 196,206 ****
          }
!         else if (pressLeft || pressRight || fire2pressed)
          {
!             if (pressLeft)
!                 arrowLefttActive = arrowActiveMax;
              
!             if (pressRight)
!                 arrowRightActive = arrowActiveMax;
              
!             itemIndex[menuSel] = !itemIndex[menuSel];
              return itemIndex[menuSel];
--- 196,209 ----
          }
!         else if (pressLeft || keyboard_check_pressed(vk_left))
          {
!             arrowLefttActive = arrowActiveMax;
              
!             itemIndex[menuSel] = max(itemIndex[menuSel] - 1, 0);
!             return itemIndex[menuSel];
!         } 
!         else if (pressRight || keyboard_check_pressed(vk_right))
!         {
!             arrowRightActive = arrowActiveMax;
              
!             itemIndex[menuSel] = min(itemIndex[menuSel] + 1, 5);
              return itemIndex[menuSel];
