*** original_scripts/gml_GlobalScript_scrSaveGame.gml	Sat Jun  7 21:06:07 2025
--- ufo50_modded_scripts/gml_GlobalScript_scrSaveGame.gml	Sun Jun  8 13:21:28 2025
***************
*** 27,32 ****
              scrWriteReal("newGame2P", true);
-     }
      
      var _makeBackup = false;
      
      if (global.backupTimer >= global.BACKUP_MINIMUM_TIME && global.currFile >= 1 && global.currFile <= global.NUM_PROFILES_ACCESSIBLE)
--- 27,45 ----
              scrWriteReal("newGame2P", true);
      
+         if (global.numPlayers == 3)
+             scrWriteReal("newGame3P", true);
+     
+         if (global.numPlayers == 4)
+             scrWriteReal("newGame4P", true);
+ 
+         if (global.numPlayers == 5)
+             scrWriteReal("newGame5P", true);
+     
+         if (global.numPlayers == 6)
+             scrWriteReal("newGame6P", true);
+     }
      var _makeBackup = false;
      
+     scrGetRecords();
+     
      if (global.backupTimer >= global.BACKUP_MINIMUM_TIME && global.currFile >= 1 && global.currFile <= global.NUM_PROFILES_ACCESSIBLE)
