*** original_scripts/gml_GlobalScript_scrDefineMeta.gml	Fri Jun 20 15:44:19 2025
--- ufo50_modded_scripts/gml_GlobalScript_scrDefineMeta.gml	Fri Jun 20 13:00:07 2025
***************
*** 41,42 ****
--- 41,43 ----
      scr34_Meta(nGame);
+     global.mGameMenuStyle[nGame] = 19;
      scrAddPillars(global.PILLAR_QUICK, global.PILLAR_REFLEX);
***************
*** 45,46 ****
--- 46,48 ----
      scr49_Meta(nGame);
+     global.mGameMenuStyle[nGame] = 18;
      scrAddPillars(global.PILLAR_QUICK, global.PILLAR_REFLEX);
***************
*** 57,58 ****
--- 59,63 ----
      scrAddPillars(global.PILLAR_QUICK, global.PILLAR_REFLEX);
+     global.mGameCoop[nGame] = true;
+     global.mGameMultiplayer[nGame] = true;
+     global.mGameMenuStyle[nGame] = 22;
      nGame++;
***************
*** 97,98 ****
--- 102,104 ----
      scr28_Meta(nGame);
+     global.mGameMenuStyle[nGame] = 20;
      scrAddPillars(global.PILLAR_QUICK, global.PILLAR_REFLEX);
***************
*** 122,123 ****
--- 128,130 ----
      scr43_Meta(nGame);
+     global.mGameMenuStyle[nGame] = 21;
      scrAddPillars(global.PILLAR_QUICK, global.PILLAR_REFLEX);
