*** original_scripts/gml_GlobalScript_scrSaveGame.gml	Sun Apr 27 20:54:41 2025
--- ufo50_modded_scripts/gml_GlobalScript_scrSaveGame.gml	Sun Apr 27 20:58:36 2025
***************
*** 27,32 ****
              scrWriteReal("newGame2P", true);
-     }
      
!     var _makeBackup = false;
      
      if (global.backupTimer >= global.BACKUP_MINIMUM_TIME && global.currFile >= 1 && global.currFile <= global.NUM_PROFILES_ACCESSIBLE)
--- 27,36 ----
              scrWriteReal("newGame2P", true);
      
!         if (global.numPlayers == 3)
!             scrWriteReal("newGame3P", true);
      
+         if (global.numPlayers == 4)
+             scrWriteReal("newGame4P", true);
+     }
+     var _makeBackup = false;
      if (global.backupTimer >= global.BACKUP_MINIMUM_TIME && global.currFile >= 1 && global.currFile <= global.NUM_PROFILES_ACCESSIBLE)
