*** original_scripts/gml_Object_oPauseMenu_Other_13.gml	Mon Sep  8 19:19:27 2025
--- ufo50_modded_scripts/gml_Object_oPauseMenu_Other_13.gml	Tue Sep  9 20:42:48 2025
***************
*** 12,14 ****
      
!     for (var p = 0; p <= 1; p++)
      {
--- 12,14 ----
      
!     for (var p = 0; p <= 5; p++)
      {
***************
*** 40,41 ****
--- 40,45 ----
      gamepadConnect[1] = gamepad_is_connected(global.joySlot[1]);
+     gamepadConnect[2] = gamepad_is_connected(global.joySlot[2]);
+     gamepadConnect[3] = gamepad_is_connected(global.joySlot[3]);
+     gamepadConnect[4] = gamepad_is_connected(global.joySlot[4]);
+     gamepadConnect[5] = gamepad_is_connected(global.joySlot[5]);
      drawMenu = true;
***************
*** 237,239 ****
                  
!                 for (var p = 0; p <= 1; p++)
                  {
--- 241,243 ----
                  
!                 for (var p = 0; p <= 5; p++)
                  {
***************
*** 262,264 ****
      {
!         for (var p = 0; p <= 1; p++)
          {
--- 266,268 ----
      {
!         for (var p = 0; p <= 5; p++)
          {
***************
*** 268,270 ****
          
!         for (var p = 0; p <= 1; p++)
          {
--- 272,274 ----
          
!         for (var p = 0; p <= 5; p++)
          {
***************
*** 278,279 ****
--- 282,287 ----
          global.joyType[1] = scrDetectJoyType(1);
+         global.joyType[2] = scrDetectJoyType(2);
+         global.joyType[3] = scrDetectJoyType(3);
+         global.joyType[4] = scrDetectJoyType(4);
+         global.joyType[5] = scrDetectJoyType(5);
          stateCounter = 1;
