*** original_scripts/gml_Object_oTitleScreens_Other_12.gml	Fri May 16 21:04:40 2025
--- ufo50_modded_scripts/gml_Object_oTitleScreens_Other_12.gml	Fri May 16 20:48:13 2025
***************
*** 78,79 ****
--- 78,81 ----
          menuOption[1] = scrStringManual("title_2p_battle", 0);
+         menuOption[2] = "3P BATTLE";
+         menuOption[3] = "4P BATTLE";
      }
***************
*** 149,152 ****
          menuOption[1] = scrStringManual("title_2p_continue", 0);
!         menuOption[2] = scrStringManual("title_1p_new_game", 0);
!         menuOption[3] = scrStringManual("title_2p_new_game", 0);
          
--- 151,163 ----
          menuOption[1] = scrStringManual("title_2p_continue", 0);
!         menuOption[2] = "3P CONTINUE";
!         menuOption[3] = "4P CONTINUE";
!         menuOption[4] = scrStringManual("title_1p_new_game", 0);
!         menuOption[5] = scrStringManual("title_2p_new_game", 0);
!         menuOption[6] = "3P NEW GAME";
!         menuOption[7] = "4P NEW GAME";
!         
!         if (newGameStarted)
!             menuSel = 0;
!         else
!             menuSel = 4;
          
***************
*** 158,163 ****
          
!         if (newGameStarted)
!             menuSel = 0;
!         else
!             menuSel = 2;
      }
--- 169,175 ----
          
!         if (!newGame3PStarted)
!             menuAvail[2] = false;
!         
!         if (!newGame4PStarted)
!             menuAvail[3] = false;
      }
***************
*** 174,175 ****
--- 186,194 ----
      }
+     else if (menuStyle == 18)
+     {
+         menuOption[0] = scrStringManual("title_1player", 0);
+         menuOption[1] = scrStringManual("title_2players", 0);
+         menuOption[2] = "4 PLAYER TEAMS";
+         menuOption[3] = scrStringManual("title_high_scores", 0);
+     }
      else
***************
*** 293,297 ****
                  scrSwitchSub(SUB_CONFIRM);
!             else if (newGameStarted && menuStyle == 15 && menuSel == 2)
                  scrSwitchSub(SUB_CONFIRM);
!             else if (newGame2PStarted && menuStyle == 15 && menuSel == 3)
                  scrSwitchSub(SUB_CONFIRM);
--- 312,320 ----
                  scrSwitchSub(SUB_CONFIRM);
!             else if (newGameStarted && menuStyle == 15 && menuSel == 4)
!                 scrSwitchSub(SUB_CONFIRM);
!             else if (newGame2PStarted && menuStyle == 15 && menuSel == 5)
!                 scrSwitchSub(SUB_CONFIRM);
!             else if (newGame3PStarted && menuStyle == 15 && menuSel == 6)
                  scrSwitchSub(SUB_CONFIRM);
!             else if (newGame4PStarted && menuStyle == 15 && menuSel == 7)
                  scrSwitchSub(SUB_CONFIRM);
***************
*** 301,302 ****
--- 324,327 ----
                  scrSwitchState(STATE_HIGHSCORE);
+             else if (menuStyle == 18 && menuSel == 3)
+                 scrSwitchState(STATE_HIGHSCORE);
              else if (menuStyle == 10 && menuSel == 3)
***************
*** 378,379 ****
--- 403,410 ----
              global.numPlayers = 2;
+ 
+         if (menuStyle == 18 && menuSel == 1)
+             global.numPlayers = 2;
+         
+         if (menuStyle == 18 && menuSel == 2)
+             global.numPlayers = 4;
          
***************
*** 387,388 ****
--- 418,425 ----
              global.numPlayers = 2;
+ 
+         if (menuStyle == 7 && menuSel == 2)
+             global.numPlayers = 3;
+ 
+         if (menuStyle == 7 && menuSel == 3)
+             global.numPlayers = 4;
          
***************
*** 406,407 ****
--- 443,450 ----
          
+         if (menuStyle == 16 && menuSel == 1)
+             global.numPlayers = 2;
+         
+         if (menuStyle == 17 && menuSel == 1)
+             global.numPlayers = 4;
+ 
          if (menuStyle == 15 && menuSel == 1)
***************
*** 409,417 ****
          
          if (menuStyle == 15 && menuSel == 3)
!             global.numPlayers = 2;
          
!         if (menuStyle == 16 && menuSel == 1)
              global.numPlayers = 2;
          
!         if (menuStyle == 17 && menuSel == 1)
              global.numPlayers = 4;
--- 452,466 ----
          
+         if (menuStyle == 15 && menuSel == 2)
+             global.numPlayers = 3;
+ 
          if (menuStyle == 15 && menuSel == 3)
!             global.numPlayers = 4;
          
!         if (menuStyle == 15 && menuSel == 5)
              global.numPlayers = 2;
          
!         if (menuStyle == 15 && menuSel == 6)
!             global.numPlayers = 3;
!         
!         if (menuStyle == 15 && menuSel == 7)
              global.numPlayers = 4;
