*** original_scripts/gml_GlobalScript_scrInit.gml	Sun Aug 10 16:08:16 2025
--- ufo50_modded_scripts/gml_GlobalScript_scrInit.gml	Fri Jul 25 12:07:53 2025
***************
*** 2,4 ****
  {
!     global.betaVersion = "1.7.6";
      global.PLAT_STEAM = 0;
--- 2,4 ----
  {
!     global.betaVersion = "1.7.5";
      global.PLAT_STEAM = 0;
***************
*** 50,52 ****
      global.QUICK_PLAY = 0;
!     global.SKIP_INTRO = false;
      global.PROFILE_PRELOAD = 0;
--- 50,52 ----
      global.QUICK_PLAY = 0;
!     global.SKIP_INTRO = true;
      global.PROFILE_PRELOAD = 0;
