*** original_scripts/gml_Object_oTitleScreens_Create_0.gml	Sun Sep 28 16:52:16 2025
--- ufo50_modded_scripts/gml_Object_oTitleScreens_Create_0.gml	Sun Sep 28 20:31:47 2025
***************
*** 8,9 ****
--- 8,11 ----
  
+ scrDeactivateSplitScreen();
+ 
  switch (introStyle)
***************
*** 57,60 ****
  saveFileExists = scrReadSaveStatus();
! newGameStarted = scrReadNewGameStatus();
! newGame2PStarted = scrReadNewGame2PStatus();
  scrCloseCurrFile();
--- 59,66 ----
  saveFileExists = scrReadSaveStatus();
! newGameStarted = scrReadReal("newGame", false);
! newGame2PStarted = scrReadReal("newGame2P", false);
! newGame3PStarted = scrReadReal("newGame3P", false);
! newGame4PStarted = scrReadReal("newGame4P", false);
! newGame5PStarted = scrReadReal("newGame5P", false);
! newGame6PStarted = scrReadReal("newGame6P", false);
  scrCloseCurrFile();
