*** original_scripts/gml_GlobalScript_scrGetInputAll.gml	Thu Jan  1 00:00:00 1970
--- ufo50_modded_scripts/gml_GlobalScript_scrGetInputAll.gml	Fri May 30 10:52:24 2025
***************
*** 0 ****
--- 1,22 ----
+ function scrGetInputAll(numPlayers)
+ {
+     for (var p = 0; p < numPlayers; p++)
+     {
+         scrGetInput(p);
+         bHoldUp[p] = holdUp;
+         bHoldDown[p] = holdDown;
+         bHoldLeft[p] = holdLeft;
+         bHoldRight[p] = holdRight;
+         bPressUp[p] = pressUp;
+         bPressDown[p] = pressDown;
+         bPressLeft[p] = pressLeft;
+         bPressRight[p] = pressRight;
+         bFire1[p] = fire1;
+         bFire2[p] = fire2;
+         bFire1pressed[p] = fire1pressed;
+         bFire2pressed[p] = fire2pressed;
+         bFire1released[p] = fire1released;
+         bFire2released[p] = fire2released;
+         bPressStart[p] = pressStart;
+     }
+ }
