*** original_scripts/gml_GlobalScript_scrDefineMeta.gml	Mon Aug 18 14:47:08 2025
--- ufo50_modded_scripts/gml_GlobalScript_scrDefineMeta.gml	Mon Aug 18 11:28:17 2025
***************
*** 41,42 ****
--- 41,43 ----
      scr34_Meta(nGame);
+     global.mGameMenuStyle[nGame] = 19;
      scrAddPillars(global.PILLAR_QUICK, global.PILLAR_REFLEX);
***************
*** 45,46 ****
--- 46,48 ----
      scr49_Meta(nGame);
+     global.mGameMenuStyle[nGame] = 18;
      scrAddPillars(global.PILLAR_QUICK, global.PILLAR_REFLEX);
***************
*** 57,58 ****
--- 59,71 ----
      scrAddPillars(global.PILLAR_QUICK, global.PILLAR_REFLEX);
+     global.mGameCoop[nGame] = true;
+     global.mGameMultiplayer[nGame] = true;
+     global.mGameMenuStyle[nGame] = 22;
+     global.g15_threshY = 96;
+     global.g15_skipOptionsMenu = false;
+     global.g15_gameOverCondition = 0;
+     global.g15_playerScores = [0, 0, 0, 0];
+     global.g15_coopCostMultiplier = 1.2;
+     global.g15_isVsMode = false;
+     global.g15_multiplayerStartingLevel = 1;
+     global.g15_randomiseLevel = false;
      nGame++;
***************
*** 64,65 ****
--- 77,79 ----
      global.mGameVersus[nGame] = true;
+     global.mGameMenuStyle[nGame] = 23;
      nGame++;
***************
*** 97,98 ****
--- 111,113 ----
      scr28_Meta(nGame);
+     global.mGameMenuStyle[nGame] = 20;
      scrAddPillars(global.PILLAR_QUICK, global.PILLAR_REFLEX);
***************
*** 122,123 ****
--- 137,139 ----
      scr43_Meta(nGame);
+     global.mGameMenuStyle[nGame] = 21;
      scrAddPillars(global.PILLAR_QUICK, global.PILLAR_REFLEX);
