*** original_scripts/the_big_bell_race/gml_Object_o28_Mas_Create_0.gml	Tue Oct 14 20:52:30 2025
--- ufo50_modded_scripts/the_big_bell_race/gml_Object_o28_Mas_Create_0.gml	Thu Oct 16 08:31:49 2025
***************
*** 11,13 ****
  
! if (global.numPlayers == 2)
      multip = true;
--- 11,13 ----
  
! if (global.numPlayers >= 2)
      multip = true;
***************
*** 15,19 ****
  navi[0] = 0;
  
  if (multip)
!     navi[1] = 1;
  else
--- 15,22 ----
  navi[0] = 0;
+ readies = 0;
  
  if (multip)
!     for (var i = 0; i < global.numPlayers; i++) {
!         navi[i] = i;
!     }
  else
***************
*** 21,26 ****
  
! naviSel[0] = 0;
! naviSel[1] = 0;
! playerSel[0] = 0;
! playerSel[1] = 1;
  cb = 0;
--- 24,29 ----
  
! for (var i = 0; i < global.numPlayers; i++) {
!     naviSel[i] = 0;
!     playerSel[i] = i;
! }
  cb = 0;
***************
*** 45,48 ****
  goalPos = 0;
! p1InGoal = false;
! p2InGoal = false;
  dsRank = ds_grid_create(2, 6);
--- 48,52 ----
  goalPos = 0;
! for (var i = 0; i < global.numPlayers; i++) {
!     playerInGoal[i] = false;
! }
  dsRank = ds_grid_create(2, 6);
***************
*** 70,72 ****
          spawn[i] = instance_nearest(x, y, o28_Spawn);
!         x += 384;
      }
--- 74,76 ----
          spawn[i] = instance_nearest(x, y, o28_Spawn);
!         x += __view_get(e__VW.WView, 0);
      }
***************
*** 143,144 ****
  tBGM = 0;
! trigEndPhaseBGM = false;
--- 147,148 ----
  tBGM = 0;
! trigEndPhaseBGM = false;
\ No newline at end of file
