*** original_scripts/the_big_bell_race/gml_Object_o28_Mas_Step_0.gml	Sat Jun  7 21:06:07 2025
--- ufo50_modded_scripts/the_big_bell_race/gml_Object_o28_Mas_Step_0.gml	Sun Jun  8 14:40:17 2025
***************
*** 5,74 ****
  {
!     scrGetInput(0);
!     
!     if (naviSel[0] == 0)
!     {
!         if (pressLeft)
!         {
!             scrSfx(sfx_navi00, 40);
!             var np = navi[0];
!             
!             while (navi[0] == np || navi[0] == navi[1])
!             {
!                 navi[0]--;
!                 
!                 if (navi[0] < 0)
!                     navi[0] = 5;
!             }
!         }
!         else if (pressRight)
!         {
!             scrSfx(sfx_navi00, 40);
!             var np = navi[0];
!             
!             while (navi[0] == np || navi[0] == navi[1])
!             {
!                 navi[0]++;
!                 
!                 if (navi[0] > 5)
!                     navi[0] = 0;
!             }
!         }
!         else if (fire2pressed)
!         {
!             if (naviSel[0] == 0)
!             {
!                 mute(sfx_deselect00);
!                 scrSfx(sfx_select01, 45);
!                 naviSel[0] = 1;
!             }
!         }
!     }
!     else if (fire1pressed)
!     {
!         if (naviSel[0] == 1)
!         {
!             mute(sfx_select01);
!             scrSfx(sfx_deselect00, 45);
!             naviSel[0] = 0;
!             t = 0;
!         }
!     }
!     
!     scrGetInput(1);
!     
!     if (global.numPlayers == 1)
!     {
!         if (fire2pressed)
!         {
!             fire2pressed = false;
!             global.numPlayers = 2;
!             multip = true;
!             navi[1] = 1;
!             scrSfx(sfx_special05, 51);
!         }
!     }
!     
!     if (multip)
!     {
!         if (naviSel[1] == 0)
          {
--- 5,10 ----
  {
!     for (var i = 0; i < global.numPlayers; i++) {
!         scrGetInput(i);
!         
!         if (naviSel[i] == 0)
          {
***************
*** 77,87 ****
                  scrSfx(sfx_navi00, 40);
!                 var np = navi[1];
!                 
!                 while (navi[1] == np || navi[1] == navi[0])
!                 {
!                     navi[1]--;
!                     
!                     if (navi[1] < 0)
!                         navi[1] = 5;
!                 }
              }
--- 13,17 ----
                  scrSfx(sfx_navi00, 40);
!                 navi[i]--;
!                 if (navi[i] < 0)
!                     navi[i] = 5;
              }
***************
*** 90,100 ****
                  scrSfx(sfx_navi00, 40);
!                 var np = navi[1];
!                 
!                 while (navi[1] == np || navi[1] == navi[0])
!                 {
!                     navi[1]++;
!                     
!                     if (navi[1] > 5)
!                         navi[1] = 0;
!                 }
              }
--- 20,24 ----
                  scrSfx(sfx_navi00, 40);
!                 navi[i]++;
!                 if (navi[i] > 5)
!                     navi[i] = 0;
              }
***************
*** 102,108 ****
              {
!                 if (naviSel[1] == 0)
                  {
                      mute(sfx_deselect00);
                      scrSfx(sfx_select01, 45);
!                     naviSel[1] = 1;
                  }
--- 26,41 ----
              {
!                 selectablePlayerFound = true;
!                     
!                 for (var j = 0; j < global.numPlayers; j++) {
!                     if (i != j && navi[i] == navi[j] && naviSel[j] == 1) {
!                         selectablePlayerFound = false
!                     }
!                 }
! 
!                 if (selectablePlayerFound && naviSel[i] == 0)
                  {
+                     readies++;
                      mute(sfx_deselect00);
                      scrSfx(sfx_select01, 45);
!                     naviSel[i] = 1;
                  }
***************
*** 112,118 ****
          {
!             if (naviSel[1] == 1)
              {
                  mute(sfx_select01);
                  scrSfx(sfx_deselect00, 45);
!                 naviSel[1] = 0;
                  t = 0;
--- 45,52 ----
          {
!             if (naviSel[i] == 1)
              {
+                 readies--;
                  mute(sfx_select01);
                  scrSfx(sfx_deselect00, 45);
!                 naviSel[i] = 0;
                  t = 0;
***************
*** 124,126 ****
      {
!         if (!multip || naviSel[1] == 1)
          {
--- 58,60 ----
      {
!         if (!multip || readies == global.numPlayers)
          {
***************
*** 131,144 ****
                  playerSel[1] = navi[1];
!                 ds_grid_set(dsScore, 0, 5, playerSel[0]);
!                 
!                 if (multip)
!                 {
!                     ds_grid_set(dsScore, 0, 4, playerSel[1]);
!                     nShuf = 3;
!                 }
!                 else
!                 {
!                     while (ds_grid_get(dsScore, 0, 4) == ds_grid_get(dsScore, 0, 5))
!                         ds_grid_set(dsScore, 0, 4, scrIRandomRange(0, 5));
                  }
                  
--- 65,74 ----
                  playerSel[1] = navi[1];
! 
!                 for (var j = 0; j < global.numPlayers; j++) {
!                     playerSel[j] = navi[j];
!                     ds_grid_set(dsScore, 0, 5 - j, playerSel[j]);
                  }
+ 
+                 while (ds_grid_get(dsScore, 0, 4) == ds_grid_get(dsScore, 0, 5))
+                     ds_grid_set(dsScore, 0, 4, scrIRandomRange(0, 5));
                  
***************
*** 318,319 ****
--- 248,269 ----
              }
+             else if (global.numPlayers >= 3 && uId == playerSel[2])
+             {
+                 racer[rCount].human = true;
+                 racer[rCount].num = 2;
+             }
+             else if (global.numPlayers >= 4 && uId == playerSel[3])
+             {
+                 racer[rCount].human = true;
+                 racer[rCount].num = 3;
+             }
+             else if (global.numPlayers >= 5 && uId == playerSel[4])
+             {
+                 racer[rCount].human = true;
+                 racer[rCount].num = 4;
+             }
+             else if (global.numPlayers >= 6 && uId == playerSel[5])
+             {
+                 racer[rCount].human = true;
+                 racer[rCount].num = 5;
+             }
              else
***************
*** 680,683 ****
  {
!     p1InGoal = false;
!     p2InGoal = false;
      goalPos = 0;
--- 630,633 ----
  {
!     playerInGoal[0] = false;
!     playerInGoal[1] = false;
      goalPos = 0;
***************
*** 688,690 ****
  }
! else if ((!multip && p1InGoal) || (multip && p1InGoal && p2InGoal))
  {
--- 638,640 ----
  }
! else if ((!multip && playerInGoal[0]) || (global.numPlayers == 2 && playerInGoal[0] && playerInGoal[1]))
  {
