*** original_scripts/the_big_bell_race/gml_GlobalScript_scr28_Checkpoints.gml	Mon Sep  8 19:19:28 2025
--- ufo50_modded_scripts/the_big_bell_race/gml_GlobalScript_scr28_Checkpoints.gml	Mon Sep  1 20:01:11 2025
***************
*** 98,107 ****
                  {
!                     if (num == 0)
!                         dad.p1InGoal = true;
!                     else
!                         dad.p2InGoal = true;
!                     
                      with (dad)
                      {
!                         if ((!multip && p1InGoal) || (multip && p1InGoal && p2InGoal))
                              scrBGM(bgm28_results);
--- 98,117 ----
                  {
!                     for (var i = 0; i < global.numPlayers; i++) {
!                         if (num == i) {
!                             dad.playerInGoal[i] = true;
!                         }
!                     }
! 
                      with (dad)
                      {
!                         var allInGoal = true;
!                         
!                         for (var i = 0; i < global.numPlayers; i++) {
!                             if (!playerInGoal[i]) {
!                                 allInGoal = false;
!                                 break;
!                             }
!                         }
! 
!                         if (allInGoal)
                              scrBGM(bgm28_results);
