*** original_scripts/attactics/gml_Object_o02_Game_Other_11.gml	Sat Jun  7 21:06:07 2025
--- ufo50_modded_scripts/attactics/gml_Object_o02_Game_Other_11.gml	Sat Jun  7 20:48:03 2025
***************
*** 56,58 ****
      
!     if (pressDown && players == 2 && lvl == TOTAL_LEVELS)
      {
--- 56,58 ----
      
!     if (pressDown && players >= 2 && lvl == TOTAL_LEVELS)
      {
***************
*** 81,84 ****
      
!     if (players == 2 && lvl == (TOTAL_LEVELS - 1))
      {
          if (fire1pressed || fire2pressed)
--- 81,86 ----
      
!     if (players >= 2 && lvl == (TOTAL_LEVELS - 1))
      {
+         scrGetInput(ControllerMap[0]);
+ 
          if (fire1pressed || fire2pressed)
***************
*** 194,196 ****
          {
!             scrGetInput(1);
              
--- 196,198 ----
          {
!             scrGetInput(ControllerMap[1]);
              
