*** original_scripts/attactics/gml_Object_o02_Game_Create_0.gml	Sat Jun  7 21:06:07 2025
--- ufo50_modded_scripts/attactics/gml_Object_o02_Game_Create_0.gml	Sat Jun  7 20:58:31 2025
***************
*** 16,17 ****
--- 16,18 ----
  STATE_PUZZLE_SETUP = 8;
+ STATE_VS_OPTIONS = 9;
  TIMER_LENGTH = 200;
***************
*** 40,41 ****
--- 41,45 ----
  players = global.numPlayers;
+ titleY = 0;
+ ControllerMap = [0, 1, 2, 3]
+ RANDOMISE_CONTROLLERS = 0;
  lvl = 1;
***************
*** 55,58 ****
  }
! else
  {
      state = STATE_LEVEL_SELECT;
--- 59,66 ----
  }
! else if (players = 4)
  {
+     state = STATE_VS_OPTIONS;
+     mode = MODE_CAMPAIGN;
+ }
+ else {
      state = STATE_LEVEL_SELECT;
***************
*** 96,105 ****
  
! for (var p = 1; p <= 2; p++)
  {
!     if (p == 1)
          selX[p] = 64;
!     else
          selX[p] = 304;
      
!     selY[p] = 114;
      selSprite[p] = s02_SelectorOff;
--- 104,117 ----
  
! for (var p = 1; p <= global.numPlayers; p++)
  {
!     if (p == 1 || p == 3)
          selX[p] = 64;
!     else if (p == 2 || p == 4)
          selX[p] = 304;
      
!     if (p <= 2)
!         selY[p] = 114;
!     else 
!         selY[p] = 146;
! 
      selSprite[p] = s02_SelectorOff;
