*** original_scripts/attactics/gml_Object_o02_Game_Step_0.gml	Sat Jun  7 20:20:13 2025
--- ufo50_modded_scripts/attactics/gml_Object_o02_Game_Step_0.gml	Sat Jun  7 20:14:16 2025
***************
*** 24,26 ****
  
! scrGetInputBoth();
  
--- 24,26 ----
  
! scrGetInputAll(players);
  
***************
*** 46,47 ****
--- 46,48 ----
          var i = p - 1;
+         var t = ((p - 1) mod 2) + 1;
          
***************
*** 73,77 ****
              
!             if (p == RED && selX[p] > 160)
                  selX[p] = 160;
!             else if (p == BLUE && selX[p] < 192)
                  selX[p] = 192;
--- 74,78 ----
              
!             if (t == RED && selX[p] > 160)
                  selX[p] = 160;
!             else if (t == BLUE && selX[p] < 192)
                  selX[p] = 192;
