*** original_scripts/velgress/gml_Object_o15_RoomGen_Alarm_0.gml	Mon Aug 18 14:47:09 2025
--- ufo50_modded_scripts/velgress/gml_Object_o15_RoomGen_Alarm_0.gml	Mon Aug 18 10:02:08 2025
***************
*** 378,379 ****
--- 378,395 ----
      }
+ 
+     if (life == 30) {
+         var xpos = playX + 104;
+         var ypos = y + 160;
+         var rock = instance_create(xpos, ypos, o15_PlatRock);
+         rock.life = 10000;
+         var rock = instance_create(xpos, ypos + 16, o15_PlatRock);
+         rock.life = 10000;
+ 
+         xpos = playX + 258;
+         ypos = y + 160;
+         var rock = instance_create(xpos, ypos, o15_PlatRock);
+         rock.life = 10000;
+         var rock = instance_create(xpos, ypos + 16, o15_PlatRock);
+         rock.life = 10000;
+     }
  }
