*** original_scripts/velgress/gml_Object_o15_Player_Step_0.gml	Mon Aug 18 14:47:09 2025
--- ufo50_modded_scripts/velgress/gml_Object_o15_Player_Step_0.gml	Mon Aug 18 10:53:25 2025
***************
*** 1,2 ****
! if (instance_exists(oTextBox) || o15_Game.state == -2 || o15_Game.state == 4)
      exit;
--- 1,7 ----
! if (instance_exists(oTextBox) 
!     || o15_Game.state == -2 
!     || o15_Game.state == o15_Game.STATE_VS_OPTIONS 
!     || o15_Game.state == o15_Game.STATE_INTRO_FALL 
!     || o15_Game.state == o15_Game.STATE_INTRO_FALL_WAIT 
!     || o15_Game.state == 4)
      exit;
***************
*** 11,13 ****
      image_speed = 0.2;
!     sprite_index = s15_Player1Walk;
      
--- 16,18 ----
      image_speed = 0.2;
!     sprite_index = s15_Player1WalkMulti;
      
***************
*** 56,58 ****
              if (o15_Game.key1 && o15_Game.key2 && o15_Game.key3)
!                 sprite_index = s15_Player1Stand;
              
--- 61,63 ----
              if (o15_Game.key1 && o15_Game.key2 && o15_Game.key3)
!                 sprite_index = s15_Player1StandMulti;
              
***************
*** 75,77 ****
              scrBGM(bgm00_alpha_stingWin, false);
!             sprite_index = s15_Player1Stand;
              scrDrawTextBoxSplit(scrString("you_did_it"));
--- 80,82 ----
              scrBGM(bgm00_alpha_stingWin, false);
!             sprite_index = s15_Player1StandMulti;
              scrDrawTextBoxSplit(scrString("you_did_it"));
***************
*** 90,92 ****
              scrBGM(bgm00_alpha_stingWin, false);
!             sprite_index = s15_Player1Stand;
              scrDrawTextBoxSplit(scrString("level_complete"));
--- 95,97 ----
              scrBGM(bgm00_alpha_stingWin, false);
!             sprite_index = s15_Player1StandMulti;
              scrDrawTextBoxSplit(scrString("level_complete"));
***************
*** 188,191 ****
          o15_Game.state = 0;
!         x = 416 + (384 * (o15_Game.level - 1)) + 8;
!         y = o15_Game.LEVEL_HEIGHT + 8;
          
--- 193,210 ----
          o15_Game.state = 0;
! 
!         with (o15_Player) {
!             if (playerNum == 1) {
!                 x = 416 + (384 * (o15_Game.level - 1)) + 8;
!             } else if (playerNum == 2) {
!                 x = 416 + (384 * (o15_Game.level - 1)) + 8 + 304;
!             }
!             else if (playerNum == 3) {
!                 x = 416 + (384 * (o15_Game.level - 1)) + 8 + 72;
!             }
!             else if (playerNum == 4) {
!                 x = 416 + (384 * (o15_Game.level - 1)) + 8 + 224;
!             }
!             y = o15_Game.LEVEL_HEIGHT + 8;
!             dead = false;
!         }
          
***************
*** 195,196 ****
--- 214,219 ----
              y = o15_Game.LEVEL_HEIGHT + 40;
+             if (global.numPlayers >= 2) {
+                 x = 396 + (384 * (o15_Game.level - 1));
+                 sprite_index = s15_SpikeRollerLarge;
+             }
          }
***************
*** 224,226 ****
  {
!     sprite_index = s15_Player1Dead;
      hspd = 0;
--- 247,249 ----
  {
!     sprite_index = s15_Player1DeadMulti;
      hspd = 0;
***************
*** 232,235 ****
  {
!     instance_create(scrViewX() + rand(64, 320), scrViewY() - 16, o15_Cavian);
!     createdBoss = true;
  }
--- 255,260 ----
  {
!     if (!instance_exists(o15_Cavian)) {
!         instance_create(scrViewX() + rand(64, 320), scrViewY() - 16, o15_Cavian);
!         createdBoss = true;
!     }
  }
***************
*** 237,240 ****
  {
!     instance_create(scrViewX() + rand(64, 320), scrViewY() - 16, o15_Cavian);
!     createdBoss = true;
  }
--- 262,267 ----
  {
!     if (!instance_exists(o15_Cavian)) {
!         instance_create(scrViewX() + rand(64, 320), scrViewY() - 16, o15_Cavian);
!         createdBoss = true;
!     }
  }
***************
*** 242,245 ****
  {
!     instance_create(scrViewX() + rand(64, 320), scrViewY() - 16, o15_Cavian);
!     createdBoss = true;
  }
--- 269,274 ----
  {
!     if (!instance_exists(o15_Cavian)) {
!         instance_create(scrViewX() + rand(64, 320), scrViewY() - 16, o15_Cavian);
!         createdBoss = true;
!     }
  }
***************
*** 251,255 ****
  
! threshY = 96;
! 
! if (y < (scrViewY() + threshY))
  {
--- 280,282 ----
  
! if (y < (scrViewY() + global.g15_threshY))
  {
***************
*** 260,262 ****
      {
!         __view_set(e__VW.YView, 0, y - threshY);
          
--- 287,289 ----
      {
!         __view_set(e__VW.YView, 0, y - global.g15_threshY);
          
***************
*** 267,269 ****
          {
!             if (o15_Game.level == 3 && sprite_index == s15_SpikeRoller)
              {
--- 294,296 ----
          {
!             if (o15_Game.level == 3 && (sprite_index == s15_SpikeRoller || sprite_index == s15_SpikeRollerLarge))
              {
***************
*** 279,281 ****
              
!             sprite_index = s15_SpikeRollerRoll;
          }
--- 306,311 ----
              
!             if (global.numPlayers >= 2)
!                 sprite_index = s15_SpikeRollerRollLarge;
!             else
!                 sprite_index = s15_SpikeRollerRoll;
          }
***************
*** 322,324 ****
  if (o15_Game.state != 3 && o15_Game.state != 4)
!     scrGetInput(0);
  
--- 352,354 ----
  if (o15_Game.state != 3 && o15_Game.state != 4)
!     scrGetInput(playerNum - 1);
  
***************
*** 448,454 ****
              {
!                 scrSfx(sfx_warp00, 80);
!                 instance_create(x, y - 12, o15_HighJumpBlast);
!                 highJumpCount = highJumpCountMax;
!                 highJumpMult = 4;
!                 alarm[0] = 1;
              }
--- 478,496 ----
              {
!                 with (o15_Player) {
!                     if (global.numPlayers >= 2) {
!                         scrSfx(sfx_warp00, 80);
!                         highBlast = instance_create(x, y - 12, o15_HighJumpBlast);
!                         highBlast.playerNum = playerNum;
!                         highJumpCount = highJumpCountMax;
!                         highJumpMult = 4;
!                         alarm[0] = 1;
!                     } else {
!                         scrSfx(sfx_warp00, 80);
!                         highBlast = instance_create(x, y - 12, o15_HighJumpBlast);
!                         highBlast.playerNum = playerNum;
!                         highJumpCount = highJumpCountMax;
!                         highJumpMult = 2;
!                         alarm[0] = 1;
!                     }
!                 }
              }
***************
*** 487,489 ****
      
!     if (fire2pressed)
      {
--- 529,531 ----
      
!     if (fire2pressed && (global.numPlayers == 1 || highJumpCount == 0))
      {
***************
*** 635,639 ****
          if (doubleJump > 0)
!             sprite_index = s15_Player1Walk;
          else
!             sprite_index = s15_Player2Walk;
      }
--- 677,681 ----
          if (doubleJump > 0)
!             sprite_index = s15_Player1WalkMulti;
          else
!             sprite_index = s15_Player2WalkMulti;
      }
***************
*** 641,643 ****
      {
!         sprite_index = s15_Player1Stand;
      }
--- 683,685 ----
      {
!         sprite_index = s15_Player1StandMulti;
      }
***************
*** 645,647 ****
      {
!         sprite_index = s15_Player2Stand;
      }
--- 687,689 ----
      {
!         sprite_index = s15_Player2StandMulti;
      }
***************
*** 650,652 ****
  {
!     sprite_index = s15_Player2Flip;
  }
--- 692,694 ----
  {
!     sprite_index = s15_Player2FlipMulti;
  }
***************
*** 657,661 ****
          if (doubleJump > 0)
!             sprite_index = s15_Player1Jump;
          else
!             sprite_index = s15_Player2Jump;
      }
--- 699,703 ----
          if (doubleJump > 0)
!             sprite_index = s15_Player1JumpMulti;
          else
!             sprite_index = s15_Player2JumpMulti;
      }
***************
*** 663,665 ****
      {
!         sprite_index = s15_Player1Fall;
      }
--- 705,707 ----
      {
!         sprite_index = s15_Player1FallMulti;
      }
***************
*** 667,669 ****
      {
!         sprite_index = s15_Player2Fall;
      }
--- 709,711 ----
      {
!         sprite_index = s15_Player2FallMulti;
      }
