*** original_scripts/velgress/gml_Object_o15_Game_Create_0.gml	Fri Jun 20 22:40:40 2025
--- ufo50_modded_scripts/velgress/gml_Object_o15_Game_Create_0.gml	Fri Jun 20 21:52:31 2025
***************
*** 2,3 ****
--- 2,7 ----
  state = -2;
+ STATE_INTRO_FALL_WAIT = -3;
+ STATE_INTRO_FALL = -4;
+ STATE_VS_OPTIONS = -5;
+ titleY = 0;
  substate = 0;
***************
*** 37,38 ****
--- 41,60 ----
  
+ numPlayers = global.numPlayers;
+ 
+ COOP_COST_MULTIPLIER = 1.2;
+ 
+ if (numPlayers >= 2) {
+     for (var i = 0; i <= 8; i++) {
+         shopPrices[i] *= COOP_COST_MULTIPLIER;
+     }
+ 
+     with (o15_SpikeRoller)
+     {
+         sprite_index = s15_SpikeRollerLarge;
+         x -= 20;
+     }
+ }
+ 
+ players[0] = o15_Player;
+ 
  for (var i = 0; i < 50; i++)
***************
*** 122,125 ****
  
! if (!inRecordOrPlayback())
!     instance_create(192, -32, o15_IntroFall);
  
--- 144,152 ----
  
! if (numPlayers == 1)
!     state = STATE_INTRO_FALL;
! else if (variable_instance_exists(global, "skipOptionsMenu") && global.skipOptionsMenu == true) {
!     global.skipOptionsMenu = false;
!     state = STATE_INTRO_FALL;
! } else
!     state = STATE_VS_OPTIONS;
  
