*** original_scripts/velgress/gml_Object_o15_Player_Step_0.gml	Fri Jun 20 22:40:40 2025
--- ufo50_modded_scripts/velgress/gml_Object_o15_Player_Step_0.gml	Sat Jun 21 18:10:38 2025
***************
*** 1,2 ****
! if (instance_exists(oTextBox) || o15_Game.state == -2 || o15_Game.state == 4)
      exit;
--- 1,7 ----
! if (instance_exists(oTextBox) 
!     || o15_Game.state == -2 
!     || o15_Game.state == o15_Game.STATE_VS_OPTIONS 
!     || o15_Game.state == o15_Game.STATE_INTRO_FALL 
!     || o15_Game.state == o15_Game.STATE_INTRO_FALL_WAIT 
!     || o15_Game.state == 4)
      exit;
***************
*** 188,191 ****
          o15_Game.state = 0;
!         x = 416 + (384 * (o15_Game.level - 1)) + 8;
!         y = o15_Game.LEVEL_HEIGHT + 8;
          
--- 193,200 ----
          o15_Game.state = 0;
! 
!         with (o15_Player) {
!             x = 416 + (384 * (o15_Game.level - 1)) + 8;
!             y = o15_Game.LEVEL_HEIGHT + 8;
!             dead = false;
!         }
          
***************
*** 195,196 ****
--- 204,209 ----
              y = o15_Game.LEVEL_HEIGHT + 40;
+             if (global.numPlayers >= 2) {
+                 x = 396 + (384 * (o15_Game.level - 1));
+                 sprite_index = s15_SpikeRollerLarge;
+             }
          }
***************
*** 251,255 ****
  
! threshY = 96;
! 
! if (y < (scrViewY() + threshY))
  {
--- 264,266 ----
  
! if (y < (scrViewY() + global.g15_threshY))
  {
***************
*** 260,262 ****
      {
!         __view_set(e__VW.YView, 0, y - threshY);
          
--- 271,273 ----
      {
!         __view_set(e__VW.YView, 0, y - global.g15_threshY);
          
***************
*** 267,269 ****
          {
!             if (o15_Game.level == 3 && sprite_index == s15_SpikeRoller)
              {
--- 278,280 ----
          {
!             if (o15_Game.level == 3 && (sprite_index == s15_SpikeRoller || sprite_index == s15_SpikeRollerLarge))
              {
***************
*** 279,281 ****
              
!             sprite_index = s15_SpikeRollerRoll;
          }
--- 290,295 ----
              
!             if (global.numPlayers >= 2)
!                 sprite_index = s15_SpikeRollerRollLarge;
!             else
!                 sprite_index = s15_SpikeRollerRoll;
          }
***************
*** 322,324 ****
  if (o15_Game.state != 3 && o15_Game.state != 4)
!     scrGetInput(0);
  
--- 336,338 ----
  if (o15_Game.state != 3 && o15_Game.state != 4)
!     scrGetInput(playerNum - 1);
  
