*** original_scripts/velgress/gml_Object_o15_Player_Collision_o15__SpikeRoller.gml	Fri Jun 20 22:40:40 2025
--- ufo50_modded_scripts/velgress/gml_Object_o15_Player_Collision_o15__SpikeRoller.gml	Sat Jun 21 20:58:18 2025
***************
*** 2,4 ****
  {
-     muteBGM();
      scrSfx(sfx_die01, 85);
--- 2,3 ----
***************
*** 11,19 ****
      vspd = 0;
-     o15_Game.state = -99;
-     o15_Game.alarm[0] = 10;
      
!     if (o15_Game.currHighFloor > o15_Game.highFloor)
!         o15_Game.newHighFloor = true;
!     
!     scrSaveGame(0);
  }
--- 10,57 ----
      vspd = 0;
      
!     var gameOver = true;
! 
!     if (global.g15_isVsMode) {
!         var playersAlive = global.numPlayers;
!         var livingPlayerNum = 1;
!         gameOver = false;
! 
!         for (var p = 0; p < global.numPlayers; p++) {
!             if (o15_Game.players[p].dead) {
!                 playersAlive--;
!             } else {
!                 livingPlayerNum = p + 1;
!             }
!         }
! 
!         if (playersAlive == 1) {
!             gameOver = true;
!             global.g15_winningPlayerNum = livingPlayerNum;
!         }
!     }
!     else if (global.g15_gameOverCondition == 0) {
!         for (var p = 0; p < global.numPlayers; p++) {
!             if (!o15_Game.players[p].dead) {
!                 gameOver = false;
!             }
!         }
!     } else if (global.g15_gameOverCondition == 1) {
!         gameOver = false;
!         for (var p = 0; p < global.numPlayers; p++) {
!             if (!o15_Game.players[p].dead) {
!                 gameOver = true;
!             }
!         }
!     }
! 
!     if (gameOver) {
!         muteBGM();
!         o15_Game.state = o15_Game.STATE_GAME_OVER;
!         o15_Game.alarm[0] = 10;
!         
!         if (o15_Game.currHighFloor > o15_Game.highFloor)
!             o15_Game.newHighFloor = true;
!         
!         scrSaveGame(0);
!     }
  }
