*** original_scripts/velgress/gml_Object_o15_Player_Step_0.gml	Tue Oct 14 20:52:30 2025
--- ufo50_modded_scripts/velgress/gml_Object_o15_Player_Step_0.gml	Thu Oct 16 08:40:25 2025
***************
*** 1,2 ****
! if (instance_exists(oTextBox) || o15_Game.state == -2 || o15_Game.state == 4)
      exit;
--- 1,7 ----
! if (instance_exists(oTextBox) 
!     || o15_Game.state == -2 
!     || o15_Game.state == o15_Game.STATE_MULTIPLAYER_OPTIONS 
!     || o15_Game.state == o15_Game.STATE_INTRO_FALL 
!     || o15_Game.state == o15_Game.STATE_INTRO_FALL_WAIT 
!     || o15_Game.state == 4)
      exit;
***************
*** 11,13 ****
      image_speed = 0.2;
!     sprite_index = s15_Player1Walk;
      
--- 16,18 ----
      image_speed = 0.2;
!     sprite_index = s15_Player1WalkMulti;
      
***************
*** 56,58 ****
              if (o15_Game.key1 && o15_Game.key2 && o15_Game.key3)
!                 sprite_index = s15_Player1Stand;
              
--- 61,63 ----
              if (o15_Game.key1 && o15_Game.key2 && o15_Game.key3)
!                 sprite_index = s15_Player1StandMulti;
              
***************
*** 75,77 ****
              scrBGM(bgm00_alpha_stingWin, false);
!             sprite_index = s15_Player1Stand;
              scrDrawTextBoxSplit(scrString("you_did_it"));
--- 80,82 ----
              scrBGM(bgm00_alpha_stingWin, false);
!             sprite_index = s15_Player1StandMulti;
              scrDrawTextBoxSplit(scrString("you_did_it"));
***************
*** 90,92 ****
              scrBGM(bgm00_alpha_stingWin, false);
!             sprite_index = s15_Player1Stand;
              scrDrawTextBoxSplit(scrString("level_complete"));
--- 95,97 ----
              scrBGM(bgm00_alpha_stingWin, false);
!             sprite_index = s15_Player1StandMulti;
              scrDrawTextBoxSplit(scrString("level_complete"));
***************
*** 187,190 ****
          o15_Game.state = 0;
!         x = 416 + (384 * (o15_Game.level - 1)) + 8;
!         y = o15_Game.LEVEL_HEIGHT + 8;
          
--- 192,209 ----
          o15_Game.state = 0;
! 
!         with (o15_Player) {
!             if (playerNum == 1) {
!                 x = 416 + (384 * (o15_Game.level - 1)) + 8;
!             } else if (playerNum == 2) {
!                 x = 416 + (384 * (o15_Game.level - 1)) + 8 + 304;
!             }
!             else if (playerNum == 3) {
!                 x = 416 + (384 * (o15_Game.level - 1)) + 8 + 72;
!             }
!             else if (playerNum == 4) {
!                 x = 416 + (384 * (o15_Game.level - 1)) + 8 + 224;
!             }
!             y = o15_Game.LEVEL_HEIGHT + 8;
!             dead = false;
!         }
          
***************
*** 194,195 ****
--- 213,218 ----
              y = o15_Game.LEVEL_HEIGHT + 40;
+             if (global.numPlayers >= 2) {
+                 x = 396 + (384 * (o15_Game.level - 1));
+                 sprite_index = s15_SpikeRollerLarge;
+             }
          }
***************
*** 229,231 ****
  {
!     sprite_index = s15_Player1Dead;
      hspd = 0;
--- 252,254 ----
  {
!     sprite_index = s15_Player1DeadMulti;
      hspd = 0;
***************
*** 268,270 ****
          if (camera_get_view_y(view_get_camera(0)) < o15_Game.borderY)
!             set_yview(o15_Game.borderY);
          
--- 291,293 ----
          if (camera_get_view_y(view_get_camera(0)) < o15_Game.borderY)
!            set_yview(o15_Game.borderY);
          
***************
*** 272,274 ****
          {
!             if (o15_Game.level == 3 && sprite_index == s15_SpikeRoller)
              {
--- 295,297 ----
          {
!             if (o15_Game.level == 3 && (sprite_index == s15_SpikeRoller || sprite_index == s15_SpikeRollerLarge))
              {
***************
*** 284,286 ****
              
!             sprite_index = s15_SpikeRollerRoll;
          }
--- 307,312 ----
              
!             if (global.numPlayers >= 2)
!                 sprite_index = s15_SpikeRollerRollLarge;
!             else
!                 sprite_index = s15_SpikeRollerRoll;
          }
***************
*** 327,329 ****
  if (o15_Game.state != 3 && o15_Game.state != 4)
!     scrGetInput(0);
  
--- 353,355 ----
  if (o15_Game.state != 3 && o15_Game.state != 4)
!     scrGetInput(playerNum - 1);
  
***************
*** 453,459 ****
              {
!                 scrSfx(sfx_warp00, 80);
!                 instance_create(x, y - 12, o15_HighJumpBlast);
!                 highJumpCount = highJumpCountMax;
!                 highJumpMult = 4;
!                 alarm[0] = 1;
              }
--- 479,488 ----
              {
!                 with (o15_Player) {
!                     scrSfx(sfx_warp00, 80);
!                     highBlast = instance_create(x, y - 12, o15_HighJumpBlast);
!                     highBlast.playerNum = playerNum;
!                     highJumpCount = highJumpCountMax;
!                     highJumpMult = 4;
!                     alarm[0] = 1;
!                 }
              }
***************
*** 492,494 ****
      
!     if (fire2pressed)
      {
--- 521,523 ----
      
!     if (fire2pressed && (global.numPlayers == 1 || highJumpCount == 0))
      {
***************
*** 640,644 ****
          if (doubleJump > 0)
!             sprite_index = s15_Player1Walk;
          else
!             sprite_index = s15_Player2Walk;
      }
--- 669,673 ----
          if (doubleJump > 0)
!             sprite_index = s15_Player1WalkMulti;
          else
!             sprite_index = s15_Player2WalkMulti;
      }
***************
*** 646,648 ****
      {
!         sprite_index = s15_Player1Stand;
      }
--- 675,677 ----
      {
!         sprite_index = s15_Player1StandMulti;
      }
***************
*** 650,652 ****
      {
!         sprite_index = s15_Player2Stand;
      }
--- 679,681 ----
      {
!         sprite_index = s15_Player2StandMulti;
      }
***************
*** 655,657 ****
  {
!     sprite_index = s15_Player2Flip;
  }
--- 684,686 ----
  {
!     sprite_index = s15_Player2FlipMulti;
  }
***************
*** 662,666 ****
          if (doubleJump > 0)
!             sprite_index = s15_Player1Jump;
          else
!             sprite_index = s15_Player2Jump;
      }
--- 691,695 ----
          if (doubleJump > 0)
!             sprite_index = s15_Player1JumpMulti;
          else
!             sprite_index = s15_Player2JumpMulti;
      }
***************
*** 668,670 ****
      {
!         sprite_index = s15_Player1Fall;
      }
--- 697,699 ----
      {
!         sprite_index = s15_Player1FallMulti;
      }
***************
*** 672,675 ****
      {
!         sprite_index = s15_Player2Fall;
      }
! }
--- 701,704 ----
      {
!         sprite_index = s15_Player2FallMulti;
      }
! }
\ No newline at end of file
