*** original_scripts/velgress/gml_Object_o15_Game_Draw_0.gml	Mon Aug 18 14:47:09 2025
--- ufo50_modded_scripts/velgress/gml_Object_o15_Game_Draw_0.gml	Mon Aug 18 11:00:28 2025
***************
*** 6,8 ****
  
! if ((state == 0 || state == -99) && instance_exists(o15_Player))
  {
--- 6,8 ----
  
! if ((state == 0 || state == STATE_GAME_OVER) && instance_exists(o15_Player))
  {
***************
*** 158,160 ****
                  draw_sprite(s15_ShopItems, shopItems[i + ((level - 1) * 3)], _xv + 32 + (i * 64), _yv + 80);
!                 draw_text(_xv + 32 + (i * 64) + 16, _yv + 80 + 24, string(shopPrices[i + ((level - 1) * 3)]));
              }
--- 158,163 ----
                  draw_sprite(s15_ShopItems, shopItems[i + ((level - 1) * 3)], _xv + 32 + (i * 64), _yv + 80);
!                 draw_text(
!                     _xv + 32 + (i * 64) + 16, _yv + 80 + 24, 
!                     string(ceil(shopPrices[i + ((level - 1) * 3)] * global.g15_coopCostMultiplier))
!                     );
              }
***************
*** 175,180 ****
  }
! else if (state == -99)
  {
      scrSetFont(global.fontTall);
!     draw_text_centered(_xv + 192, _yv + 96, scrStringExt("game_over", "*", 42, 4), 8);
      scrSetFont(global.fontDefault);
--- 178,203 ----
  }
! else if (state == STATE_GAME_OVER)
  {
      scrSetFont(global.fontTall);
!     if (global.g15_isVsMode) {
!         draw_text_centered(_xv + 192, _yv + 30, "PLAYER " + string(global.g15_winningPlayerNum) + " WINS", 8);
! 
!         draw_sprite(s15_Player1TorsoMulti, 0, _xv + 166, _yv + 106);
!         draw_text_centered(_xv + 218, _yv + 96, string(global.g15_playerScores[0]) + " WINS", 8);
! 
!         draw_sprite(s15_Player1TorsoMulti, 1, _xv + 166, _yv + 126);
!         draw_text_centered(_xv + 218, _yv + 116, string(global.g15_playerScores[1]) + " WINS", 8);
! 
!         if (global.numPlayers >= 3) {
!             draw_sprite(s15_Player1TorsoMulti, 2, _xv + 166, _yv + 146);
!             draw_text_centered(_xv + 218, _yv + 136, string(global.g15_playerScores[2]) + " WINS", 8);
!         }
! 
!         if (global.numPlayers >= 4) {
!             draw_sprite(s15_Player1TorsoMulti, 3, _xv + 166, _yv + 166);
!             draw_text_centered(_xv + 218, _yv + 156, string(global.g15_playerScores[3]) + " WINS", 8);
!         }
!     } else {
!         draw_text_centered(_xv + 192, _yv + 96, scrStringExt("game_over", "*", 42, 4), 8);
!     }
      scrSetFont(global.fontDefault);
***************
*** 189 ****
--- 212,259 ----
  }
+ 
+ 
+ if (state == STATE_VS_OPTIONS)
+ {
+     draw_set_color(global.palette[31]);
+     draw_set_alpha(0.65);
+     draw_rectangle(0, 0, 384, 240, false);
+     draw_set_alpha(1);
+     draw_set_color(c_white);
+     scrSetFont(global.fontTall);
+     draw_text(100, 24, "VS OPTIONS");
+     scrSetFont(global.fontDefault);
+ 
+     draw_text(50, 80, "PLAYERS:");
+     var controllerText = string(global.numPlayers);
+     draw_text(210, 80, controllerText);
+ 
+     draw_text(50, 96, "SCROLL THRESHOLD:");
+     var scrollThresholdText = threshYOptions[currentThreshYSelect];
+     draw_text(210, 96, scrollThresholdText);
+ 
+     draw_text(50, 112, "GAME OVER WHEN:");
+     if (global.g15_isVsMode) {
+         var gameOverConditionText = "LAST PERSON STANDING";
+     } else {
+         var gameOverConditionText = gameOverConditions[currentGameOverConditionSelect];
+     }
+     draw_text(210, 112, gameOverConditionText);
+ 
+     draw_text(50, 128, "STARTING LEVEL:");
+     var startingLevelText = string(startingLevelSelection);
+     if (startingLevelSelection == 5) {
+         startingLevelText = "RANDOM LEVEL";
+     }
+     draw_text(210, 128, startingLevelText);
+ 
+     draw_text(50, 144, "ITEM COST:");
+     var coopCostText = coopCostOptions[coopCostSelection];
+     draw_text(210, 144, coopCostText);
+ 
+     draw_text(50, 160, "START");
+     
+     if (scrTransitionDone())
+     {
+         draw_sprite_ext(s15_Keys, 0, 34, 83 + (titleY * 16), 1, 1, 0, c_white, 1);
+     }
+ }
\ No newline at end of file
