*** original_scripts/velgress/gml_Object_o15_RoomGen_Alarm_0.gml	Mon Sep  8 19:19:28 2025
--- ufo50_modded_scripts/velgress/gml_Object_o15_RoomGen_Alarm_0.gml	Tue Sep  9 21:54:43 2025
***************
*** 178,180 ****
              {
!                 if (rand(1, bounceChance) == 1)
                  {
--- 178,180 ----
              {
!                 if (global.numPlayers == 1 && rand(1, bounceChance) == 1)
                  {
***************
*** 199,201 ****
              {
!                 if (rand(1, bounceChance) == 1)
                  {
--- 199,201 ----
              {
!                 if (global.numPlayers == 1 && rand(1, bounceChance) == 1)
                  {
***************
*** 224,226 ****
              {
!                 if (rand(1, bounceChance) == 1)
                  {
--- 224,226 ----
              {
!                 if (global.numPlayers == 1 && rand(1, bounceChance) == 1)
                  {
***************
*** 335,337 ****
          
!         if (bounceChance > 1)
              bounceChance -= 1;
--- 335,337 ----
          
!         if (global.numPlayers == 1 && bounceChance > 1)
              bounceChance -= 1;
***************
*** 377,378 ****
--- 377,394 ----
      }
+ 
+     if (life == 30) {
+         var xpos = playX + 104;
+         var ypos = y + 160;
+         var rock = instance_create(xpos, ypos, o15_PlatRock);
+         rock.life = 10000;
+         var rock = instance_create(xpos, ypos + 16, o15_PlatRock);
+         rock.life = 10000;
+ 
+         xpos = playX + 258;
+         ypos = y + 160;
+         var rock = instance_create(xpos, ypos, o15_PlatRock);
+         rock.life = 10000;
+         var rock = instance_create(xpos, ypos + 16, o15_PlatRock);
+         rock.life = 10000;
+     }
  }
***************
*** 399,400 ****
--- 415,423 ----
  
+ with (o15__Platform)
+     visible = true;
+     
+ with (o15__JumpThru)
+     visible = true;
+ 
+ 
  instance_destroy();
