*** original_scripts/velgress/gml_Object_o15_RoomGen_Alarm_0.gml	Fri Aug 22 11:55:09 2025
--- ufo50_modded_scripts/velgress/gml_Object_o15_RoomGen_Alarm_0.gml	Fri Aug 22 11:18:31 2025
***************
*** 179,181 ****
              {
!                 if (rand(1, bounceChance) == 1)
                  {
--- 179,181 ----
              {
!                 if (global.numPlayers == 1 && rand(1, bounceChance) == 1)
                  {
***************
*** 200,202 ****
              {
!                 if (rand(1, bounceChance) == 1)
                  {
--- 200,202 ----
              {
!                 if (global.numPlayers == 1 && rand(1, bounceChance) == 1)
                  {
***************
*** 225,227 ****
              {
!                 if (rand(1, bounceChance) == 1)
                  {
--- 225,227 ----
              {
!                 if (global.numPlayers == 1 && rand(1, bounceChance) == 1)
                  {
***************
*** 336,338 ****
          
!         if (bounceChance > 1)
              bounceChance -= 1;
--- 336,338 ----
          
!         if (global.numPlayers == 1 && bounceChance > 1)
              bounceChance -= 1;
***************
*** 378,379 ****
--- 378,395 ----
      }
+ 
+     if (life == 30) {
+         var xpos = playX + 104;
+         var ypos = y + 160;
+         var rock = instance_create(xpos, ypos, o15_PlatRock);
+         rock.life = 10000;
+         var rock = instance_create(xpos, ypos + 16, o15_PlatRock);
+         rock.life = 10000;
+ 
+         xpos = playX + 258;
+         ypos = y + 160;
+         var rock = instance_create(xpos, ypos, o15_PlatRock);
+         rock.life = 10000;
+         var rock = instance_create(xpos, ypos + 16, o15_PlatRock);
+         rock.life = 10000;
+     }
  }
