*** original_scripts/velgress/gml_Object_o15_SpikeRoller_Step_0.gml	Mon Sep  8 19:19:28 2025
--- ufo50_modded_scripts/velgress/gml_Object_o15_SpikeRoller_Step_0.gml	Tue Sep  9 21:16:28 2025
***************
*** 2,3 ****
--- 2,8 ----
  {
+     if (global.numPlayers >= 2)
+         x = camera_get_view_x(view_get_camera(0)) + 12;
+     else
+         x = camera_get_view_x(view_get_camera(0)) + 32;
+ 
      x = camera_get_view_x(view_get_camera(0)) + 32;
