*** original_scripts/hot_foot/gml_Object_o43_Game_Other_11.gml	Sun Jun  1 20:32:04 2025
--- ufo50_modded_scripts/hot_foot/gml_Object_o43_Game_Other_11.gml	Sun Jun  1 20:28:50 2025
***************
*** 102,104 ****
      
!     if (draftTeam == 1 && playMode != DEMO)
      {
--- 102,113 ----
      
!     
!     if (draftTurn <= 2 && numPlayers == 4) {
!         draftPlayer = draftTeam;
!         inputPlayer = (draftTeam - 1) * 2 + 1;
!     }
!     else if (draftTurn >= 3 && numPlayers == 4) {
!         draftPlayer = draftTeam;
!         inputPlayer = (draftTeam - 1) * 2 + 2;
!     }
!     else if (draftTeam == 1 && playMode != DEMO)
      {
