*** original_scripts/hot_foot/gml_Object_o43_Game_Other_12.gml	Sat Jun  7 07:56:49 2025
--- ufo50_modded_scripts/hot_foot/gml_Object_o43_Game_Other_12.gml	Sat Jun  7 08:05:19 2025
***************
*** 50,51 ****
--- 50,57 ----
      draftX[1] = 12;
+ 
+     if (playMode == VS && numPlayers == 4) {
+         draftX[1] = 2;
+         draftX[2] = 11;
+         draftX[3] = 12;
+     }
      
***************
*** 60,62 ****
      
!     inputPlayer = 3;
  }
--- 66,69 ----
      
!     inputPlayer = 5;
!     scrGetInputAll(numPlayers);
  }
***************
*** 68,70 ****
      
!     for (var p = 0; p <= (numPlayers - 1); p++)
      {
--- 75,77 ----
      
!     for (var p = 0; p < numPlayers; p++)
      {
***************
*** 72,75 ****
          
!         if (playMode != COOP)
              t = p;
          else
--- 79,84 ----
          
!         if (playMode == VS && numPlayers == 2)
              t = p;
+         else if (playMode == VS && numPlayers == 4)
+             t = p div 2;
          else
***************
*** 79,81 ****
          
!         if (playMode != COOP && athlete[(2 * p) + 2])
              teamFull = true;
--- 88,92 ----
          
!         if (playMode == VS && numPlayers == 2 && athlete[(2 * p) + 2])
!             teamFull = true;
!         else if (playMode == VS && numPlayers == 4 && athlete[2 * (t + 1)] && athlete[2 * t + 1])
              teamFull = true;
***************
*** 92,94 ****
          
!         if (bPressRight[p] && !teamFull)
          {
--- 103,105 ----
          
!         if (bPressRight[ControllerMap[p] - 1] && !teamFull)
          {
***************
*** 106,108 ****
          
!         if (bPressLeft[p] && !teamFull)
          {
--- 117,119 ----
          
!         if (bPressLeft[ControllerMap[p] - 1] && !teamFull)
          {
***************
*** 128,130 ****
          
!         if (bFire2pressed[p] && !invalidPick && !teamFull)
          {
--- 139,141 ----
          
!         if (bFire2pressed[ControllerMap[p] - 1] && !invalidPick && !teamFull)
          {
***************
*** 133,135 ****
              
!             if (playMode != COOP)
              {
--- 144,146 ----
              
!             if (playMode == VS && numPlayers <= 2)
              {
***************
*** 138,140 ****
                  else if (t == 1 && !athlete[3])
!                     pick = 1;
                  else
--- 149,153 ----
                  else if (t == 1 && !athlete[3])
!                     pick = 3;
!                 else if (t == 1 && !athlete[4])
!                     pick = 4;
                  else
***************
*** 142,143 ****
--- 155,160 ----
              }
+             else if (playMode == VS && numPlayers == 4)
+             {
+                 pick = p mod 2 + 1;
+             }
              else if (playMode == COOP)
***************
*** 202,204 ****
              }
!             else if (numPlayers == 2 && playMode == VS && athlete[1] && athlete[2] && athlete[3] && athlete[4])
              {
--- 219,221 ----
              }
!             else if ((numPlayers == 2 || numPlayers = 4) && playMode == VS && athlete[1] && athlete[2] && athlete[3] && athlete[4])
              {
***************
*** 213,215 ****
          }
!         else if (bFire2pressed[p])
          {
--- 230,232 ----
          }
!         else if (bFire2pressed[ControllerMap[p] - 1])
          {
***************
*** 218,220 ****
          
!         if (bFire1pressed[p] && (athlete[(2 * t) + 1] || athlete[(2 * t) + 2]))
          {
--- 235,237 ----
          
!         if (bFire1pressed[ControllerMap[p] - 1] && (athlete[(2 * t) + 1] || athlete[(2 * t) + 2]))
          {
