*** original_scripts/hot_foot/gml_Object_o43_Game_Other_11.gml	Fri Nov  7 20:49:39 2025
--- ufo50_modded_scripts/hot_foot/gml_Object_o43_Game_Other_11.gml	Sat Dec  6 19:52:41 2025
***************
*** 12,19 ****
      lastboss = false;
-     
      for (var i = 1; i <= 14; i++)
          chars[i][PLAYED] = false;
!     
      draftOptions[1] = scrIRandomRange(1, 12);
-     
      for (var a = 2; a <= 5; a++)
--- 12,18 ----
      lastboss = false;
      for (var i = 1; i <= 14; i++)
+     {
          chars[i][PLAYED] = false;
!     }
      draftOptions[1] = scrIRandomRange(1, 12);
      for (var a = 2; a <= 5; a++)
***************
*** 21,23 ****
          var redundant;
-         
          do
--- 20,21 ----
***************
*** 27,29 ****
              var i = a - 1;
-             
              while (i >= 1)
--- 25,26 ----
***************
*** 31,34 ****
                  if (draftOptions[i] == draftOptions[a])
                      redundant = true;
!                 
                  i--;
--- 28,32 ----
                  if (draftOptions[i] == draftOptions[a])
+                 {
                      redundant = true;
!                 }
                  i--;
***************
*** 38,40 ****
      }
-     
      for (var i = 1; i <= 5; i++)
--- 36,37 ----
***************
*** 42,44 ****
          var draftable = instance_create((i * 32) + 96, 110, o43_Draftable);
-         
          if (verify(draftable))
--- 39,40 ----
***************
*** 50,77 ****
              draftIDs[i] = draftable.id;
-             
              if ((i % 2) == 0)
                  draftable.image_xscale = -1;
          }
      }
-     
      draftTeam = scrChoose(1, 2);
!     
      if (numPlayers == 1 || playMode == COOP)
          draftTeam = 1;
!     
      draftTurn = 0;
      draftX = 1;
-     
      with (o43_Coach)
          visible = false;
!     
      with (o43_Bumper)
          visible = false;
!     
      with (o43_Bumper2)
          visible = false;
!     
      metaTrigger = 0;
  }
- 
  if (subState == 1)
--- 46,82 ----
              draftIDs[i] = draftable.id;
              if ((i % 2) == 0)
+             {
                  draftable.image_xscale = -1;
+             }
          }
      }
      draftTeam = scrChoose(1, 2);
!     if (numPlayers == 4 && playMode == VS)
!     {
!         draftTeamPlayerFirst = scrChoose(0, 1);
!     }
!     else
!     {
!         draftTeamPlayerFirst = 0;
!     }
      if (numPlayers == 1 || playMode == COOP)
+     {
          draftTeam = 1;
!     }
      draftTurn = 0;
      draftX = 1;
      with (o43_Coach)
+     {
          visible = false;
!     }
      with (o43_Bumper)
+     {
          visible = false;
!     }
      with (o43_Bumper2)
+     {
          visible = false;
!     }
      metaTrigger = 0;
  }
  if (subState == 1)
***************
*** 79,81 ****
      draftTurn++;
-     
      if (draftTurn == 5)
--- 84,85 ----
***************
*** 85,87 ****
      }
-     
      while (!draftOptions[draftX])
--- 89,90 ----
***************
*** 89,104 ****
          draftX++;
-         
          if (draftX > 5)
              draftX = 1;
      }
-     
      scrInputClear();
-     
      if (draftTurn == 2)
          draftTeam = !(draftTeam - 1) + 1;
!     
      if (draftTurn == 4)
          draftTeam = !(draftTeam - 1) + 1;
!     
!     if (draftTeam == 1 && playMode != DEMO)
      {
--- 92,118 ----
          draftX++;
          if (draftX > 5)
+         {
              draftX = 1;
+         }
      }
      scrInputClear();
      if (draftTurn == 2)
+     {
          draftTeam = !(draftTeam - 1) + 1;
!     }
      if (draftTurn == 4)
+     {
          draftTeam = !(draftTeam - 1) + 1;
!     }
!     if (draftTurn <= 2 && numPlayers == 4 && playMode == VS)
!     {
!         draftPlayer = draftTeamPlayerFirst + 1;
!         inputPlayer = ControllerMap[((draftTeam - 1) * 2) + draftTeamPlayerFirst];
!     }
!     else if (draftTurn >= 3 && numPlayers == 4 && playMode == VS)
!     {
!         draftPlayer = !draftTeamPlayerFirst + 1;
!         inputPlayer = ControllerMap[((draftTeam - 1) * 2) + !draftTeamPlayerFirst];
!     }
!     else if (draftTeam == 1 && playMode != DEMO)
      {
***************
*** 125,130 ****
      }
-     
      subState = 2;
  }
- 
  if (subState == 2)
--- 139,142 ----
***************
*** 136,141 ****
          scrInputClear();
-         
          if ((aiTimer % 19) == 0)
              pressRight = true;
!         
          if ((aiTimer % 37) == 0 && stateTimer >= 64)
--- 148,153 ----
          scrInputClear();
          if ((aiTimer % 19) == 0)
+         {
              pressRight = true;
!         }
          if ((aiTimer % 37) == 0 && stateTimer >= 64)
***************
*** 146,153 ****
      }
-     
      if (draftPlayer == 2 && holdDown)
          metaTrigger++;
      else
          metaTrigger = 0;
!     
      if (pressRight)
--- 158,167 ----
      }
      if (draftPlayer == 2 && holdDown)
+     {
          metaTrigger++;
+     }
      else
+     {
          metaTrigger = 0;
!     }
      if (pressRight)
***************
*** 155,157 ****
          scrSfx(soundNaviB, 50);
-         
          do
--- 169,170 ----
***************
*** 159,163 ****
              draftX++;
-             
              if (draftX > 5)
                  draftX = 1;
          }
--- 172,177 ----
              draftX++;
              if (draftX > 5)
+             {
                  draftX = 1;
+             }
          }
***************
*** 165,167 ****
      }
-     
      if (pressLeft)
--- 179,180 ----
***************
*** 169,171 ****
          scrSfx(soundNaviB, 50);
-         
          do
--- 182,183 ----
***************
*** 173,177 ****
              draftX--;
-             
              if (draftX < 0)
                  draftX = 5;
          }
--- 185,190 ----
              draftX--;
              if (draftX < 0)
+             {
                  draftX = 5;
+             }
          }
***************
*** 179,181 ****
      }
-     
      with (o43_Draftable)
--- 192,193 ----
***************
*** 183,189 ****
          if (id == other.draftIDs[other.draftX])
              selected = true;
          else
              selected = false;
      }
-     
      if (fire2pressed)
--- 195,204 ----
          if (id == other.draftIDs[other.draftX])
+         {
              selected = true;
+         }
          else
+         {
              selected = false;
+         }
      }
      if (fire2pressed)
***************
*** 192,205 ****
          var pick;
-         
          if (draftTurn <= 2)
!             pick = 1;
          else
!             pick = 2;
!         
          if (draftTeam == 1)
              athlete[pick] = draftOptions[draftX];
!         
          if (draftTeam == 2)
              athlete[pick + 2] = draftOptions[draftX];
!         
          with (draftIDs[draftX])
--- 207,224 ----
          var pick;
          if (draftTurn <= 2)
!         {
!             pick = 1 + draftTeamPlayerFirst;
!         }
          else
!         {
!             pick = 1 + !draftTeamPlayerFirst;
!         }
          if (draftTeam == 1)
+         {
              athlete[pick] = draftOptions[draftX];
!         }
          if (draftTeam == 2)
+         {
              athlete[pick + 2] = draftOptions[draftX];
!         }
          with (draftIDs[draftX])
***************
*** 211,213 ****
              }
-             
              if (other.draftTeam == 2)
--- 230,231 ----
***************
*** 217,219 ****
              }
-             
              y = 172;
--- 235,236 ----
***************
*** 221,228 ****
          }
-         
          draftOptions[draftX] = false;
-         
          with (o43_Draftable)
              selected = false;
!         
          subState = 3;
--- 238,244 ----
          }
          draftOptions[draftX] = false;
          with (o43_Draftable)
+         {
              selected = false;
!         }
          subState = 3;
***************
*** 230,232 ****
  }
- 
  if (subState == 3)
--- 246,247 ----
***************
*** 234,236 ****
      stateTimer++;
-     
      if (stateTimer >= 48)
--- 249,250 ----
***************
*** 241,243 ****
  }
- 
  if (subState == 4)
--- 255,256 ----
***************
*** 245,247 ****
      stateTimer++;
-     
      if (stateTimer >= 24)
--- 258,259 ----
***************
*** 253,256 ****
                  with (draftIDs[i])
                      crying = true;
!                 
                  lastboss = draftOptions[i];
--- 265,269 ----
                  with (draftIDs[i])
+                 {
                      crying = true;
!                 }
                  lastboss = draftOptions[i];
***************
*** 258,260 ****
          }
-         
          stateTimer = 0;
--- 271,272 ----
***************
*** 263,265 ****
  }
- 
  if (subState == 5)
--- 275,276 ----
***************
*** 267,272 ****
      if (stateTimer == 0)
          scrSfx(soundCry, 61);
!     
      stateTimer++;
-     
      if (stateTimer >= 100)
--- 278,283 ----
      if (stateTimer == 0)
+     {
          scrSfx(soundCry, 61);
!     }
      stateTimer++;
      if (stateTimer >= 100)
***************
*** 279,281 ****
  }
- 
  if (subState == 6)
--- 290,291 ----
***************
*** 285,289 ****
--- 295,303 ----
          if (!shownTutorial)
+         {
              futureState = STATE_TUTORIAL;
+         }
          else
+         {
              futureState = STATE_MATCH_SETUP;
+         }
      }
