*** original_scripts/hot_foot/gml_Object_o43_Game_Draw_0.gml	Sat Jun  7 07:56:49 2025
--- ufo50_modded_scripts/hot_foot/gml_Object_o43_Game_Draw_0.gml	Sat Jun  7 08:12:46 2025
***************
*** 30,32 ****
          draw_set_alpha(0.65);
!         draw_rectangle(0, 0, 384, 216, false);
          draw_set_alpha(1);
--- 30,32 ----
          draw_set_alpha(0.65);
!         draw_rectangle(0, 0, 384, 240, false);
          draw_set_alpha(1);
***************
*** 41,43 ****
          scrStringDrawExt(88, 144, "option_jump", 0, 16, 0);
!         scrStringDrawExt(88, 160, "option_exit", 0, 16, 0);
          draw_text(224, 80, string(POINT_GOAL));
--- 41,44 ----
          scrStringDrawExt(88, 144, "option_jump", 0, 16, 0);
!         draw_text(88, 160, "CONTROLLERS:");
!         scrStringDrawExt(88, 176, "option_exit", 0, 16, 0);
          draw_text(224, 80, string(POINT_GOAL));
***************
*** 47,48 ****
--- 48,53 ----
          draw_text(224, 144, scrString("setting_jump_" + string(TM_JUMP_SYNC + 1)));
+         var controllerText = "FIXED"
+         if (RANDOMISE_CONTROLLERS)
+             controllerText = "RANDOM";
+         draw_text(224, 160, controllerText);
          
***************
*** 134,136 ****
          
!         if (playMode != COOP && athlete[(2 * p) + 2])
              teamFull = true;
--- 139,143 ----
          
!         if (playMode == VS && numPlayers <= 2 && athlete[(2 * p) + 2])
!             teamFull = true;
!         else if (playMode == VS && numPlayers == 4 && athlete[2 * (p div 2) + 2])
              teamFull = true;
***************
*** 143,144 ****
--- 150,152 ----
              var xShift;
+             var yShift = 0;
              
***************
*** 148,149 ****
--- 156,162 ----
                  xShift = 192 * p;
+             else if (numPlayers == 4)
+             {
+                 xShift = 192 * (p div 2);
+                 yShift = 134 * (p mod 2);
+             }
              
***************
*** 151,156 ****
              {
!                 if (flicker8 || p == 0 || draftPos != draftX[0] || athlete[2])
!                 {
!                     draw_sprite(s43_DraftPicker, p, ((draftPos * 24) + 36) - (24 * charUnlock), 144);
!                     draw_sprite(s43_DraftPicker, p + 2, ((draftPos * 24) + 36) - (24 * charUnlock), 144);
                  }
--- 164,174 ----
              {
!                 if (numPlayers != 4) {
!                     if (flicker8 || p == 0 || draftPos != draftX[0] || athlete[2])
!                     {
!                         draw_sprite(s43_DraftPicker, p, ((draftPos * 24) + 36) - (24 * charUnlock), 144);
!                         draw_sprite(s43_DraftPicker, p + 2, ((draftPos * 24) + 36) - (24 * charUnlock), 144);
!                     }
!                 } else {
!                     draw_sprite(s43_DraftPicker, p div 2, ((draftPos * 24) + 36) - (24 * charUnlock), 144);
!                     draw_sprite(s43_DraftPicker, (p mod 2) + 2, ((draftPos * 24) + 36) - (24 * charUnlock), 144);
                  }
***************
*** 166,182 ****
              
!             draw_sprite(s43_InfoBox, 0, xShift, 16);
              draw_set_color(c_white);
              scrSetFont(global.fontTall);
!             draw_text(8 + xShift, 24, chars[draftOptions[draftPos]][NAME]);
              
              if (winners[draftOptions[draftPos]])
!                 draw_sprite(s43_TrophyIcon, 0, 16 + xShift + (8 * string_length(chars[draftOptions[draftPos]][NAME])), 32);
              
              scrSetFont(global.fontDefault);
!             draw_text_ext(8 + xShift, 48, scrStringExt("special_throw", 0, 14, 0), 8, 56);
              var sp = chars[draftOptions[draftPos]][ST];
!             scr43_DrawSpecialThrowIcon(sp, 80 + xShift, 56);
!             draw_text_ext(104 + xShift, 48, scrStringExt("special_action", 0, 14, 0), 8, 56);
              var sa = chars[draftOptions[draftPos]][SA];
!             scr43_DrawSpecialActionIcon(sa, 176 + xShift, 56);
              
--- 184,200 ----
              
!             draw_sprite(s43_InfoBox, 0, xShift, 16 + yShift);
              draw_set_color(c_white);
              scrSetFont(global.fontTall);
!             draw_text(8 + xShift, 24 + yShift, chars[draftOptions[draftPos]][NAME]);
              
              if (winners[draftOptions[draftPos]])
!                 draw_sprite(s43_TrophyIcon, 0, 16 + xShift + (8 * string_length(chars[draftOptions[draftPos]][NAME])), 32 + yShift);
              
              scrSetFont(global.fontDefault);
!             draw_text_ext(8 + xShift, 48 + yShift, scrStringExt("special_throw", 0, 14, 0), 8, 56);
              var sp = chars[draftOptions[draftPos]][ST];
!             scr43_DrawSpecialThrowIcon(sp, 80 + xShift, 56 + yShift);
!             draw_text_ext(104 + xShift, 48 + yShift, scrStringExt("special_action", 0, 14, 0), 8, 56);
              var sa = chars[draftOptions[draftPos]][SA];
!             scr43_DrawSpecialActionIcon(sa, 176 + xShift, 56 + yShift);
              
***************
*** 188,190 ****
              var tr = chars[draftOptions[draftPos]][TRAIT_TEXT];
!             draw_text(8 + xShift, 72, tr);
              draw_line(8 + xShift, 40, 183 + xShift, 40);
--- 206,208 ----
              var tr = chars[draftOptions[draftPos]][TRAIT_TEXT];
!             draw_text(8 + xShift, 72 + yShift, tr);
              draw_line(8 + xShift, 40, 183 + xShift, 40);
***************
*** 268,270 ****
      }
!     else if (numPlayers == 2 && playMode == VS)
      {
--- 286,288 ----
      }
!     else if (numPlayers >= 2 && playMode == VS)
      {
***************
*** 378,380 ****
          
!         if (numPlayers == 2 && playMode == VS)
          {
--- 396,398 ----
          
!         if ((numPlayers == 2 || numPlayers == 4) && playMode == VS)
          {
