*** original_scripts/hot_foot/gml_Object_o43_Game_Other_13.gml	Sat Jun  7 07:56:49 2025
--- ufo50_modded_scripts/hot_foot/gml_Object_o43_Game_Other_13.gml	Sat Jun  7 08:08:45 2025
***************
*** 49,51 ****
      {
!         a1.controller = 1;
          a1.team = 1;
--- 49,54 ----
      {
!         if (numPlayers == 4 && playMode == VS)
!             a1.controller = ControllerMap[0];
!         else
!             a1.controller = 1;
          a1.team = 1;
***************
*** 65,66 ****
--- 68,71 ----
              a2.controller = 2;
+         else if (numPlayers == 4 && playMode == VS)
+             a2.controller = ControllerMap[1];
          else
***************
*** 83,84 ****
--- 88,91 ----
              a3.controller = 2;
+         else if (numPlayers == 4 && playMode == VS)
+             a3.controller = ControllerMap[2];
          else
***************
*** 99,101 ****
      {
!         a4.controller = 0;
          a4.team = 2;
--- 106,111 ----
      {
!         if (numPlayers == 4 && playMode == VS)
!             a4.controller = ControllerMap[3];
!         else
!             a4.controller = 0;
          a4.team = 2;
