*** original_scripts/ninpek/gml_Object_o34_Balloon_Step_0.gml	Sun May 25 20:11:30 2025
--- ufo50_modded_scripts/ninpek/gml_Object_o34_Balloon_Step_0.gml	Fri May 23 21:22:29 2025
***************
*** 5,10 ****
  {
!     if (o34_Mas.multip && (o34_Mas.life[0] < 0 || o34_Mas.life[1] < 0))
!     {
!         mode = 1;
!         sprite_index = s34_BalloonB;
      }
--- 5,14 ----
  {
!     if (global.numPlayers >= 2) {
!         for (var i = 0; i < global.numPlayers; i++) {
!             if (o34_Mas.life[i] < 0)
!             {
!                 mode = 1;
!                 sprite_index = s34_BalloonB;
!             }
!         }
      }
