*** original_scripts/ninpek/gml_Object_o34_Balloon_Step_0.gml	Tue Oct 14 20:52:30 2025
--- ufo50_modded_scripts/ninpek/gml_Object_o34_Balloon_Step_0.gml	Thu Oct 16 08:11:46 2025
***************
*** 5,10 ****
  {
!     if (o34_Mas.multip && (o34_Mas.life[0] < 0 || o34_Mas.life[1] < 0))
!     {
!         mode = 1;
!         sprite_index = s34_BalloonB;
      }
--- 5,14 ----
  {
!     if (global.numPlayers >= 2) {
!         for (var i = 0; i < global.numPlayers; i++) {
!             if (o34_Mas.life[i] < 0)
!             {
!                 mode = 1;
!                 sprite_index = s34_BalloonB;
!             }
!         }
      }
***************
*** 29,45 ****
          var _y = camera_get_view_y(view_get_camera(0)) - 48;
-         var ghost = instance_create(_x, _y, o34_Ghost);
          
!         if (snum == 0)
          {
!             o34_Mas.life[1] = 0;
!             ghost.num = 1;
!             ghost.myWpn = o34_Shuriken;
!             ghost.sprite_index = s34_P2Ghost;
!         }
!         else
!         {
!             o34_Mas.life[0] = 0;
!             ghost.num = 0;
!             ghost.myWpn = o34_Shuriken2;
!             ghost.sprite_index = s34_NinjaGhost;
          }
--- 33,60 ----
          var _y = camera_get_view_y(view_get_camera(0)) - 48;
          
!         for (var i = 0; i < global.numPlayers; i++)
          {
!             if (i != snum && o34_Mas.life[i] < 0) {
!                 var ghost = instance_create(_x, _y, o34_Ghost);
!                 o34_Mas.life[i] = 0;
!                 ghost.num = i;
! 
!                 if (i == 0) {
!                     ghost.myWpn = o34_Shuriken;
!                     ghost.sprite_index = s34_NinjaGhost;
!                     ghost.x -= 30;
!                 } else if (i == 1) {
!                     ghost.myWpn = o34_Shuriken2;
!                     ghost.sprite_index = s34_P2Ghost;
!                     ghost.x -= 10;
!                 } else if (i == 2) {
!                     ghost.myWpn = o34_Shuriken3;
!                     ghost.sprite_index = s34_P3Ghost;
!                     ghost.x += 10;
!                 } else if (i == 3) {
!                     ghost.myWpn = o34_Shuriken4;
!                     ghost.sprite_index = s34_P4Ghost;
!                     ghost.x += 20;
!                 }
!             }
          }
