*** original_scripts/ninpek/gml_Object_o34_Ghost_Step_0.gml	Tue Oct 14 20:52:30 2025
--- ufo50_modded_scripts/ninpek/gml_Object_o34_Ghost_Step_0.gml	Thu Oct 16 08:28:17 2025
***************
*** 102,107 ****
                      
!                     if (myWpn == o34_Shuriken)
                          shuri.sprite_index = s34_Gshot;
!                     else
                          shuri.sprite_index = s34_Gshot2;
                  }
--- 102,111 ----
                      
!                     if (shuri.num == 0)
                          shuri.sprite_index = s34_Gshot;
!                     else if (shuri.num == 1)
                          shuri.sprite_index = s34_Gshot2;
+                     else if (shuri.num == 2)
+                         shuri.sprite_index = s34_Gshot3;
+                     else if (shuri.num == 3)
+                         shuri.sprite_index = s34_Gshot4;
                  }
***************
*** 113,116 ****
          x = camera_get_view_x(view_get_camera(0)) + 8;
!     else if (x > ((camera_get_view_x(view_get_camera(0)) + 384) - 8))
!         x = (camera_get_view_x(view_get_camera(0)) + 384) - 8;
      
--- 117,120 ----
          x = camera_get_view_x(view_get_camera(0)) + 8;
!     else if (x > ((camera_get_view_x(view_get_camera(0)) + __view_get(e__VW.WView, 0)) - 8))
!         x = (camera_get_view_x(view_get_camera(0)) + __view_get(e__VW.WView, 0)) - 8;
      
***************
*** 118,121 ****
          y = camera_get_view_y(view_get_camera(0)) + 8;
!     else if (y > ((camera_get_view_y(view_get_camera(0)) + 216) - 16))
!         y = (camera_get_view_y(view_get_camera(0)) + 216) - 16;
      
--- 122,125 ----
          y = camera_get_view_y(view_get_camera(0)) + 8;
!     else if (y > ((camera_get_view_y(view_get_camera(0)) + __view_get(e__VW.HView, 0)) - 16))
!         y = (camera_get_view_y(view_get_camera(0)) + __view_get(e__VW.HView, 0)) - 16;
      
***************
*** 132,142 ****
          {
              spawn.spriteRun = s34_NinjaRun;
              spawn.spriteJump = s34_NinjaJump2;
-             spawn.myWpn = o34_Shuriken;
          }
!         else
          {
              spawn.spriteRun = s34_P2Run;
              spawn.spriteJump = s34_P2Jump2;
!             spawn.myWpn = o34_Shuriken2;
          }
--- 136,158 ----
          {
+             spawn.myWpn = o34_Shuriken;
              spawn.spriteRun = s34_NinjaRun;
              spawn.spriteJump = s34_NinjaJump2;
          }
!         else if (num == 1)
          {
+             spawn.myWpn = o34_Shuriken2;
              spawn.spriteRun = s34_P2Run;
              spawn.spriteJump = s34_P2Jump2;
!         }
!         else if (num == 2)
!         {
!             spawn.myWpn = o34_Shuriken3;
!             spawn.spriteRun = s34_P3Run;
!             spawn.spriteJump = s34_P3Jump2;
!         }
!         else if (num == 3)
!         {
!             spawn.myWpn = o34_Shuriken4;
!             spawn.spriteRun = s34_P4Run;
!             spawn.spriteJump = s34_P4Jump2;
          }
***************
*** 150,151 ****
      }
! }
--- 166,167 ----
      }
! }
\ No newline at end of file
