*** original_scripts/mooncat/gml_GlobalScript_scr11_SaveGame.gml	Fri Sep 19 19:34:55 2025
--- ufo50_modded_scripts/mooncat/gml_GlobalScript_scr11_SaveGame.gml	Mon Sep 15 19:12:12 2025
***************
*** 8,9 ****
--- 8,11 ----
              scrWriteReal("p2_input", global.g11_input[1]);
+             scrWriteReal("p3_input", global.g11_input[2]);
+             scrWriteReal("p4_input", global.g11_input[3]);
              var numLevelsVisited = 0;
