*** original_scripts/onion_delivery/gml_Object_o32_Mas_Create_0.gml	Tue Oct 14 20:52:30 2025
--- ufo50_modded_scripts/onion_delivery/gml_Object_o32_Mas_Create_0.gml	Sun Oct 19 20:56:57 2025
***************
*** 67,68 ****
--- 67,73 ----
  PLAYFIELD_HEIGHT = 216;
+ 
+ if (global.numPlayers >= 2) {
+     scrActivateSplitScreen();
+ }
+ 
  GARDEN_GOAL = 2;
***************
*** 71,73 ****
  generalTimer = 0;
! player = -4;
  stateBuffered = -4;
--- 76,80 ----
  generalTimer = 0;
! 
! players = [];
! 
  stateBuffered = -4;
***************
*** 104,109 ****
  tDayEnd = 0;
! currRegion = 0;
! timerSpawnSystem = 0;
! activeSpawner[0] = 0;
! activeSpawnerCount = 0;
  cameraDirection = 0;
--- 111,117 ----
  tDayEnd = 0;
! currRegion = [0, 0, 0, 0];
! timerSpawnSystem = [0, 0, 0, 0];
! 
! activeSpawner = [[], [], [], []];
! activeSpawnerCount = [0, 0, 0, 0];
  cameraDirection = 0;
***************
*** 119,122 ****
  cameraOffsetAimY = 0;
! xPlayfield = 0;
! yPlayfield = 0;
  eventNewsImage[EVENT_NOTHING] = 0;
--- 127,130 ----
  cameraOffsetAimY = 0;
! xPlayfields = [];
! yPlayfields = [];
  eventNewsImage[EVENT_NOTHING] = 0;
***************
*** 145,152 ****
  showDebugInfo = false;
! statePortrait = PORTRAIT_STATIC;
! stepPortrait = 0;
! tPortrait = 0;
! turnPortrait = 0;
  displayTimer = 0;
  displayTimerPrev = displayTimer;
  checkpointActiveName1 = "";
--- 153,165 ----
  showDebugInfo = false;
! for (var i = 0; i < global.numPlayers; i++) {
!     statePortraits[i] = PORTRAIT_STATIC;
! }
! for (var i = 0; i < global.numPlayers; i++) {
!     stepPortraits[i] = 0;
!     tPortraits[i] = 0;
!     turnPortrait[i] = 0;
! }
  displayTimer = 0;
  displayTimerPrev = displayTimer;
+ checkpointActive = 0;
  checkpointActiveName1 = "";
***************
*** 223 ****
--- 236,258 ----
  metaTrigger = false;
+ 
+ 
+ enum e__VW
+ {
+     XView,
+     YView,
+     WView,
+     HView,
+     Angle,
+     HBorder,
+     VBorder,
+     HSpeed,
+     VSpeed,
+     Object,
+     Visible,
+     XPort,
+     YPort,
+     WPort,
+     HPort,
+     Camera,
+     SurfaceID
+ }
